--- ClockSprite.java
+++ ClockSprite.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.texture;
 
 import net.minecraft.client.MinecraftClient;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class ClockSprite extends Sprite {
 	private double field_2146;
 	private double field_2147;
 
@@ -31,11 +31,11 @@
 
 			while (e >= 0.5) {
 				e--;
 			}
 
-			e = MathHelper.clamp(e, -1.0, 1.0);
+			e = Mth.clamp(e, -1.0, 1.0);
 			this.field_2147 += e * 0.1;
 			this.field_2147 *= 0.8;
 			this.field_2146 = this.field_2146 + this.field_2147;
 			int i = (int)((this.field_2146 + 1.0) * (double)this.frames.size()) % this.frames.size();
 
