--- LayeredTexture.java
+++ LayeredTexture.java
@@ -3,10 +3,15 @@
 import com.google.common.collect.Lists;
 import java.awt.image.BufferedImage;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.List;
+
+import com.mojang.blaze3d.platform.NativeImage;
+import com.mojang.blaze3d.systems.GpuDevice;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.textures.TextureFormat;
 import net.minecraft.resource.ResourceManager;
 import net.minecraft.util.Identifier;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -38,8 +43,12 @@
 		} catch (IOException var7) {
 			LOGGER.error("Couldn't load layered image", var7);
 			return;
 		}
 
-		TextureUtil.method_5858(this.getGlId(), bufferedImage);
+		NativeImage nativeImage = NativeImage.read(NativeImage.Format.BGRA, bufferedImage.getWidth(), bufferedImage.getHeight(), bufferedImage.getRGB(0, 0, bufferedImage.getWidth(), bufferedImage.getHeight(), null, 0, bufferedImage.getWidth()));
+		GpuDevice gpuDevice = RenderSystem.getDevice();
+		this.texture = gpuDevice.createTexture(this.locations::toString, 5, TextureFormat.RGBA8, nativeImage.getWidth(), nativeImage.getHeight(), 1, 1);
+		this.textureView = gpuDevice.createTextureView(this.texture);
+		gpuDevice.createCommandEncoder().writeToTexture(this.texture, nativeImage);
 	}
 }
