--- NativeImageBackedTexture.java
+++ NativeImageBackedTexture.java
@@ -1,10 +1,17 @@
 package net.minecraft.client.texture;
 
-import java.awt.image.BufferedImage;
+import java.awt.image.*;
 import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import com.mojang.blaze3d.platform.NativeImage;
+import com.mojang.blaze3d.systems.GpuDevice;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.textures.TextureFormat;
 import net.minecraft.resource.ResourceManager;
+import org.lwjgl.BufferUtils;
 
 public class NativeImageBackedTexture extends AbstractTexture {
 	private final int[] pixels;
 	private final int width;
 	private final int height;
@@ -17,19 +24,25 @@
 
 	public NativeImageBackedTexture(int i, int j) {
 		this.width = i;
 		this.height = j;
 		this.pixels = new int[i * j];
-		TextureUtil.prepareImage(this.getGlId(), i, j);
+
+		GpuDevice gpuDevice = RenderSystem.getDevice();
+		this.texture = gpuDevice.createTexture(this::toString, 5, TextureFormat.RGBA8, width, height, 1, 1);
+		this.textureView = gpuDevice.createTextureView(this.texture);
 	}
 
 	@Override
 	public void load(ResourceManager resourceManager) throws IOException {
 	}
 
 	public void upload() {
-		TextureUtil.method_5861(this.getGlId(), this.pixels, this.width, this.height);
+		assert this.texture != null;
+
+		GpuDevice gpuDevice = RenderSystem.getDevice();
+        gpuDevice.createCommandEncoder().writeToTexture(this.texture, BufferUtils.createIntBuffer(pixels.length).put(pixels).flip(), NativeImage.Format.RGB, 0, 0, width, height, 0, 0);
 	}
 
 	public int[] getPixels() {
 		return this.pixels;
 	}
