--- PlayerSkinTexture.java
+++ PlayerSkinTexture.java
@@ -86,11 +86,11 @@
 	protected void method_6993() {
 		this.field_6551 = new Thread("Texture Downloader #" + field_8080.incrementAndGet()) {
 			public void run() {
 				HttpURLConnection httpURLConnection = null;
 				PlayerSkinTexture.field_8079
-					.debug("Downloading http texture from {} to {}", new Object[]{PlayerSkinTexture.this.field_6548, PlayerSkinTexture.this.field_8081});
+						.debug("Downloading http texture from {} to {}", new Object[]{PlayerSkinTexture.this.field_6548, PlayerSkinTexture.this.field_8081});
 
 				try {
 					httpURLConnection = (HttpURLConnection)new URL(PlayerSkinTexture.this.field_6548).openConnection(MinecraftClient.getInstance().getNetworkProxy());
 					httpURLConnection.setDoInput(true);
 					httpURLConnection.setDoOutput(false);
