--- EntityNavigation.java
+++ EntityNavigation.java
@@ -59,11 +59,11 @@
 			return path;
 		}
 	}
 
 	public boolean startMovingTo(double d, double e, double f, double g) {
-		Path path = this.findPathTo((double)MathHelper.floor(d), (double)((int)e), (double)MathHelper.floor(f));
+		Path path = this.findPathTo((double) MathHelper.floor(d), (double)((int)e), (double) MathHelper.floor(f));
 		return this.startMovingAlong(path, g);
 	}
 
 	public void method_11038(float f) {
 		this.field_11967 = f;
