--- PathNodeMaker.java
+++ PathNodeMaker.java
@@ -1,10 +1,10 @@
 package net.minecraft.entity.ai.pathing;
 
 import net.minecraft.entity.Entity;
 import net.minecraft.util.collection.IntObjectStorage;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.BlockView;
 
 public abstract class PathNodeMaker {
 	protected BlockView blockView;
 	protected IntObjectStorage<PathNode> pathNodeCache = new IntObjectStorage<>();
@@ -13,13 +13,13 @@
 	protected int entityBlockZSize;
 
 	public void init(BlockView blockView, Entity entity) {
 		this.blockView = blockView;
 		this.pathNodeCache.clear();
-		this.entityBlockXSize = MathHelper.floor(entity.width + 1.0F);
-		this.entityBlockYSize = MathHelper.floor(entity.height + 1.0F);
-		this.entityBlockZSize = MathHelper.floor(entity.width + 1.0F);
+		this.entityBlockXSize = Mth.floor(entity.width + 1.0F);
+		this.entityBlockYSize = Mth.floor(entity.height + 1.0F);
+		this.entityBlockZSize = Mth.floor(entity.width + 1.0F);
 	}
 
 	public void clear() {
 	}
 
