--- PathNode.java
+++ PathNode.java
@@ -1,8 +1,8 @@
 package net.minecraft.entity.ai.pathing;
 
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class PathNode {
 	public final int posX;
 	public final int posY;
 	public final int posZ;
@@ -27,11 +27,11 @@
 
 	public float getDistance(PathNode pathNode) {
 		float f = (float)(pathNode.posX - this.posX);
 		float g = (float)(pathNode.posY - this.posY);
 		float h = (float)(pathNode.posZ - this.posZ);
-		return MathHelper.sqrt(f * f + g * g + h * h);
+		return Mth.sqrt(f * f + g * g + h * h);
 	}
 
 	public float getSquaredDistance(PathNode pathNode) {
 		float f = (float)(pathNode.posX - this.posX);
 		float g = (float)(pathNode.posY - this.posY);
