--- WaterPathNodeMaker.java
+++ WaterPathNodeMaker.java
@@ -3,11 +3,11 @@
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.Entity;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.BlockView;
 
 public class WaterPathNodeMaker extends PathNodeMaker {
 	@Override
 	public void init(BlockView blockView, Entity entity) {
@@ -20,17 +20,17 @@
 	}
 
 	@Override
 	public PathNode getStart(Entity entity) {
 		return this.getNode(
-			MathHelper.floor(entity.getBoundingBox().minX), MathHelper.floor(entity.getBoundingBox().minY + 0.5), MathHelper.floor(entity.getBoundingBox().minZ)
+			Mth.floor(entity.getBoundingBox().minX), Mth.floor(entity.getBoundingBox().minY + 0.5), Mth.floor(entity.getBoundingBox().minZ)
 		);
 	}
 
 	@Override
 	public PathNode getNode(Entity entity, double d, double e, double f) {
-		return this.getNode(MathHelper.floor(d - (double)(entity.width / 2.0F)), MathHelper.floor(e + 0.5), MathHelper.floor(f - (double)(entity.width / 2.0F)));
+		return this.getNode(Mth.floor(d - (double)(entity.width / 2.0F)), Mth.floor(e + 0.5), Mth.floor(f - (double)(entity.width / 2.0F)));
 	}
 
 	@Override
 	public int getSuccessors(PathNode[] pathNodes, Entity entity, PathNode pathNode, PathNode pathNode2, float f) {
 		int i = 0;
