--- MoveControl.java
+++ MoveControl.java
@@ -1,10 +1,10 @@
 package net.minecraft.entity.ai.control;
 
 import net.minecraft.entity.attribute.EntityAttributes;
 import net.minecraft.entity.mob.MobEntity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class MoveControl {
 	protected MobEntity entity;
 	protected double targetX;
 	protected double targetY;
@@ -37,28 +37,28 @@
 
 	public void updateMovement() {
 		this.entity.setForwardSpeed(0.0F);
 		if (this.moving) {
 			this.moving = false;
-			int i = MathHelper.floor(this.entity.getBoundingBox().minY + 0.5);
+			int i = Mth.floor(this.entity.getBoundingBox().minY + 0.5);
 			double d = this.targetX - this.entity.x;
 			double e = this.targetZ - this.entity.z;
 			double f = this.targetY - (double)i;
 			double g = d * d + f * f + e * e;
 			if (!(g < 2.5000003E-7F)) {
-				float h = (float)(MathHelper.atan2(e, d) * 180.0 / (float) Math.PI) - 90.0F;
+				float h = (float)(Mth.atan2(e, d) * 180.0 / (float) Math.PI) - 90.0F;
 				this.entity.yaw = this.wrapDegrees(this.entity.yaw, h, 30.0F);
 				this.entity.setMovementSpeed((float)(this.speed * this.entity.initializeAttribute(EntityAttributes.GENERIC_MOVEMENT_SPEED).getValue()));
 				if (f > 0.0 && d * d + e * e < 1.0) {
 					this.entity.getJumpControl().setActive();
 				}
 			}
 		}
 	}
 
 	protected float wrapDegrees(float f, float g, float h) {
-		float i = MathHelper.wrapDegrees(g - f);
+		float i = Mth.wrapDegrees(g - f);
 		if (i > h) {
 			i = h;
 		}
 
 		if (i < -h) {
