--- HealthScoreboardCriteria.java
+++ HealthScoreboardCriteria.java
@@ -1,10 +1,10 @@
 package net.minecraft.scoreboard;
 
 import java.util.List;
 import net.minecraft.entity.player.PlayerEntity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class HealthScoreboardCriteria extends GenericScoreboardCriteria {
 	public HealthScoreboardCriteria(String string) {
 		super(string);
 	}
@@ -19,11 +19,11 @@
 
 		if (list.size() > 0) {
 			f /= (float)list.size();
 		}
 
-		return MathHelper.ceil(f);
+		return Mth.ceil(f);
 	}
 
 	@Override
 	public boolean method_4919() {
 		return true;
