--- GiantSpruceTreeFeature.java
+++ GiantSpruceTreeFeature.java
@@ -9,11 +9,11 @@
 import net.minecraft.block.LeavesBlock;
 import net.minecraft.block.Log1Block;
 import net.minecraft.block.PlanksBlock;
 import net.minecraft.block.material.Material;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class GiantSpruceTreeFeature extends AbstractGiantTreeFeature {
 	private static final BlockState SPRUCE_LOGS = Blocks.LOG.getDefaultState().with(Log1Block.VARIANT, PlanksBlock.WoodType.SPRUCE);
 	private static final BlockState SPRUCE_LEAVES = Blocks.LEAVES
@@ -68,11 +68,11 @@
 		int m = random.nextInt(5) + (this.useBaseHeight ? this.baseHeight : 3);
 		int n = 0;
 
 		for (int o = k - m; o <= k; o++) {
 			int p = k - o;
-			int q = l + MathHelper.floor((float)p / (float)m * 3.5F);
+			int q = l + Mth.floor((float)p / (float)m * 3.5F);
 			this.generateLeavesLimited(world, new BlockPos(i, o, j), q + (p > 0 && q == n && (o & 1) == 0 ? 1 : 0));
 			n = q;
 		}
 	}
 
