--- IceSpikeFeature.java
+++ IceSpikeFeature.java
@@ -3,11 +3,11 @@
 import java.util.Random;
 import net.minecraft.block.Block;
 import net.minecraft.block.Blocks;
 import net.minecraft.block.material.Material;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class IceSpikeFeature extends Feature {
 	@Override
 	public boolean generate(World world, Random random, BlockPos blockPos) {
@@ -25,17 +25,17 @@
 				blockPos = blockPos.up(10 + random.nextInt(30));
 			}
 
 			for (int k = 0; k < i; k++) {
 				float f = (1.0F - (float)k / (float)i) * (float)j;
-				int l = MathHelper.ceil(f);
+				int l = Mth.ceil(f);
 
 				for (int m = -l; m <= l; m++) {
-					float g = (float)MathHelper.abs(m) - 0.25F;
+					float g = (float) Mth.abs(m) - 0.25F;
 
 					for (int n = -l; n <= l; n++) {
-						float h = (float)MathHelper.abs(n) - 0.25F;
+						float h = (float) Mth.abs(n) - 0.25F;
 						if ((m == 0 && n == 0 || !(g * g + h * h > f * f)) && (m != -l && m != l && n != -l && n != l || !(random.nextFloat() > 0.75F))) {
 							Block block = world.getBlockState(blockPos.add(m, k, n)).getBlock();
 							if (block.getMaterial() == Material.AIR || block == Blocks.DIRT || block == Blocks.SNOW || block == Blocks.ICE) {
 								this.setBlockStateWithoutUpdatingNeighbors(world, blockPos.add(m, k, n), Blocks.PACKED_ICE.getDefaultState());
 							}
