--- IceSpikeFeature.java
+++ IceSpikeFeature.java
@@ -28,14 +28,14 @@
 			for (int k = 0; k < i; k++) {
 				float f = (1.0F - (float)k / (float)i) * (float)j;
 				int l = MathHelper.ceil(f);
 
 				for (int m = -l; m <= l; m++) {
-					float g = (float)MathHelper.abs(m) - 0.25F;
+					float g = (float) MathHelper.abs(m) - 0.25F;
 
 					for (int n = -l; n <= l; n++) {
-						float h = (float)MathHelper.abs(n) - 0.25F;
+						float h = (float) MathHelper.abs(n) - 0.25F;
 						if ((m == 0 && n == 0 || !(g * g + h * h > f * f)) && (m != -l && m != l && n != -l && n != l || !(random.nextFloat() > 0.75F))) {
 							Block block = world.getBlockState(blockPos.add(m, k, n)).getBlock();
 							if (block.getMaterial() == Material.AIR || block == Blocks.DIRT || block == Blocks.SNOW || block == Blocks.ICE) {
 								this.setBlockStateWithoutUpdatingNeighbors(world, blockPos.add(m, k, n), Blocks.PACKED_ICE.getDefaultState());
 							}
