--- AbstractFluidBlock.java
+++ AbstractFluidBlock.java
@@ -10,11 +10,11 @@
 import net.minecraft.state.StateManager;
 import net.minecraft.state.property.IntProperty;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.BlockView;
 import net.minecraft.world.World;
 
 public abstract class AbstractFluidBlock extends Block {
@@ -244,11 +244,11 @@
 		}
 	}
 
 	public static double getDirection(BlockView blockView, BlockPos blockPos, Material material) {
 		Vec3d vec3d = getFlowingFluidByMaterial(material).getFluidVec(blockView, blockPos);
-		return vec3d.x == 0.0 && vec3d.z == 0.0 ? -1000.0 : MathHelper.atan2(vec3d.z, vec3d.x) - (Math.PI / 2);
+		return vec3d.x == 0.0 && vec3d.z == 0.0 ? -1000.0 : Mth.atan2(vec3d.z, vec3d.x) - (Math.PI / 2);
 	}
 
 	@Override
 	public void onCreation(World world, BlockPos blockPos, BlockState blockState) {
 		this.canChangeFromLava(world, blockPos, blockState);
