--- ChestBlock.java
+++ ChestBlock.java
@@ -18,11 +18,11 @@
 import net.minecraft.state.property.DirectionProperty;
 import net.minecraft.util.ItemScatterer;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.BlockView;
 import net.minecraft.world.World;
 
 public class ChestBlock extends BlockWithEntity {
 	public static final DirectionProperty FACING = DirectionProperty.of("facing", Direction.DirectionType.HORIZONTAL);
@@ -84,11 +84,11 @@
 		return this.getDefaultState().with(FACING, livingEntity.getHorizontalDirection());
 	}
 
 	@Override
 	public void onPlaced(World world, BlockPos blockPos, BlockState blockState, LivingEntity livingEntity, ItemStack itemStack) {
-		Direction direction = Direction.fromHorizontal(MathHelper.floor((double)(livingEntity.yaw * 4.0F / 360.0F) + 0.5) & 3).getOpposite();
+		Direction direction = Direction.fromHorizontal(Mth.floor((double)(livingEntity.yaw * 4.0F / 360.0F) + 0.5) & 3).getOpposite();
 		blockState = blockState.with(FACING, direction);
 		BlockPos blockPos2 = blockPos.north();
 		BlockPos blockPos3 = blockPos.south();
 		BlockPos blockPos4 = blockPos.west();
 		BlockPos blockPos5 = blockPos.east();
@@ -387,11 +387,11 @@
 			BlockEntity blockEntity = blockView.getBlockEntity(blockPos);
 			if (blockEntity instanceof ChestBlockEntity) {
 				i = ((ChestBlockEntity)blockEntity).viewerCount;
 			}
 
-			return MathHelper.clamp(i, 0, 15);
+			return Mth.clamp(i, 0, 15);
 		}
 	}
 
 	@Override
 	public int getStrongRedstonePower(BlockView blockView, BlockPos blockPos, BlockState blockState, Direction direction) {
