--- CropBlock.java
+++ CropBlock.java
@@ -5,11 +5,11 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
 import net.minecraft.state.StateManager;
 import net.minecraft.state.property.IntProperty;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class CropBlock extends PlantBlock implements Growable {
 	public static final IntProperty AGE = IntProperty.of("age", 0, 7);
 
@@ -42,11 +42,11 @@
 			}
 		}
 	}
 
 	public void applyGrowth(World world, BlockPos blockPos, BlockState blockState) {
-		int i = (Integer)blockState.get(AGE) + MathHelper.nextInt(world.random, 2, 5);
+		int i = (Integer)blockState.get(AGE) + Mth.nextInt(world.random, 2, 5);
 		if (i > 7) {
 			i = 7;
 		}
 
 		world.setBlockState(blockPos, blockState.with(AGE, i), 2);
