--- DaylightDetectorBlock.java
+++ DaylightDetectorBlock.java
@@ -11,11 +11,11 @@
 import net.minecraft.item.itemgroup.ItemGroup;
 import net.minecraft.state.StateManager;
 import net.minecraft.state.property.IntProperty;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.BlockView;
 import net.minecraft.world.LightType;
 import net.minecraft.world.World;
 
 public class DaylightDetectorBlock extends BlockWithEntity {
@@ -48,12 +48,12 @@
 			BlockState blockState = world.getBlockState(blockPos);
 			int i = world.getLightAtPos(LightType.SKY, blockPos) - world.getAmbientDarkness();
 			float f = world.getSkyAngleRadians(1.0F);
 			float g = f < (float) Math.PI ? 0.0F : (float) (Math.PI * 2);
 			f += (g - f) * 0.2F;
-			i = Math.round((float)i * MathHelper.cos(f));
-			i = MathHelper.clamp(i, 0, 15);
+			i = Math.round((float)i * Mth.cos(f));
+			i = Mth.clamp(i, 0, 15);
 			if (this.inverted) {
 				i = 15 - i;
 			}
 
 			if ((Integer)blockState.get(POWER) != i) {
