--- OreBlock.java
+++ OreBlock.java
@@ -6,11 +6,11 @@
 import net.minecraft.item.Item;
 import net.minecraft.item.Items;
 import net.minecraft.item.itemgroup.ItemGroup;
 import net.minecraft.util.DyeColor;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class OreBlock extends Block {
 	public OreBlock() {
 		this(Material.STONE.getColor());
@@ -59,19 +59,19 @@
 	public void randomDropAsItem(World world, BlockPos blockPos, BlockState blockState, float f, int i) {
 		super.randomDropAsItem(world, blockPos, blockState, f, i);
 		if (this.getDropItem(blockState, world.random, i) != Item.fromBlock(this)) {
 			int j = 0;
 			if (this == Blocks.COAL_ORE) {
-				j = MathHelper.nextInt(world.random, 0, 2);
+				j = Mth.nextInt(world.random, 0, 2);
 			} else if (this == Blocks.DIAMOND_ORE) {
-				j = MathHelper.nextInt(world.random, 3, 7);
+				j = Mth.nextInt(world.random, 3, 7);
 			} else if (this == Blocks.EMERALD_ORE) {
-				j = MathHelper.nextInt(world.random, 3, 7);
+				j = Mth.nextInt(world.random, 3, 7);
 			} else if (this == Blocks.LAPIS_LAZULI_ORE) {
-				j = MathHelper.nextInt(world.random, 2, 5);
+				j = Mth.nextInt(world.random, 2, 5);
 			} else if (this == Blocks.NETHER_QUARTZ_ORE) {
-				j = MathHelper.nextInt(world.random, 2, 5);
+				j = Mth.nextInt(world.random, 2, 5);
 			}
 
 			this.dropExperience(world, blockPos, j);
 		}
 	}
