--- PistonBlock.java
+++ PistonBlock.java
@@ -13,11 +13,11 @@
 import net.minecraft.state.property.BooleanProperty;
 import net.minecraft.state.property.DirectionProperty;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.BlockView;
 import net.minecraft.world.World;
 
 public class PistonBlock extends Block {
 	public static final DirectionProperty DIRECTION = DirectionProperty.of("facing");
@@ -240,11 +240,11 @@
 		int j = i & 7;
 		return j > 5 ? null : Direction.getById(j);
 	}
 
 	public static Direction getDirectionFromEntity(World world, BlockPos blockPos, LivingEntity livingEntity) {
-		if (MathHelper.abs((float)livingEntity.x - (float)blockPos.getX()) < 2.0F && MathHelper.abs((float)livingEntity.z - (float)blockPos.getZ()) < 2.0F) {
+		if (Mth.abs((float)livingEntity.x - (float)blockPos.getX()) < 2.0F && Mth.abs((float)livingEntity.z - (float)blockPos.getZ()) < 2.0F) {
 			double d = livingEntity.y + (double)livingEntity.getEyeHeight();
 			if (d - (double)blockPos.getY() > 2.0) {
 				return Direction.UP;
 			}
 
