--- WeightedPressurePlateBlock.java
+++ WeightedPressurePlateBlock.java
@@ -4,11 +4,11 @@
 import net.minecraft.block.material.MaterialColor;
 import net.minecraft.entity.Entity;
 import net.minecraft.state.StateManager;
 import net.minecraft.state.property.IntProperty;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class WeightedPressurePlateBlock extends AbstractPressurePlateBlock {
 	public static final IntProperty POWER = IntProperty.of("power", 0, 15);
 	private final int weight;
@@ -26,11 +26,11 @@
 	@Override
 	protected int getRedstoneOutput(World world, BlockPos blockPos) {
 		int i = Math.min(world.getEntitiesInBox(Entity.class, this.getPlateHitBox(blockPos)).size(), this.weight);
 		if (i > 0) {
 			float f = (float)Math.min(this.weight, i) / (float)this.weight;
-			return MathHelper.ceil(f * 15.0F);
+			return Mth.ceil(f * 15.0F);
 		} else {
 			return 0;
 		}
 	}
 
