--- RealmsSliderButton.java
+++ RealmsSliderButton.java
@@ -1,10 +1,10 @@
 package net.minecraft.realms;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.MinecraftClient;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class RealmsSliderButton extends RealmsButton {
 	public float value = 1.0F;
 	public boolean sliding;
 	private final float minValue;
@@ -26,20 +26,20 @@
 	public String getMessage() {
 		return "";
 	}
 
 	public float toPct(float f) {
-		return MathHelper.clamp((this.clamp(f) - this.minValue) / (this.maxValue - this.minValue), 0.0F, 1.0F);
+		return Mth.clamp((this.clamp(f) - this.minValue) / (this.maxValue - this.minValue), 0.0F, 1.0F);
 	}
 
 	public float toValue(float f) {
-		return this.clamp(this.minValue + (this.maxValue - this.minValue) * MathHelper.clamp(f, 0.0F, 1.0F));
+		return this.clamp(this.minValue + (this.maxValue - this.minValue) * Mth.clamp(f, 0.0F, 1.0F));
 	}
 
 	public float clamp(float f) {
 		f = this.clampSteps(f);
-		return MathHelper.clamp(f, this.minValue, this.maxValue);
+		return Mth.clamp(f, this.minValue, this.maxValue);
 	}
 
 	protected float clampSteps(float f) {
 		if (this.steps > 0) {
 			f = (float)(this.steps * Math.round(f / (float)this.steps));
@@ -56,11 +56,11 @@
 	@Override
 	public void renderBg(int i, int j) {
 		if (this.getProxy().visible) {
 			if (this.sliding) {
 				this.value = (float)(i - (this.getProxy().x + 4)) / (float)(this.getProxy().getWidth() - 8);
-				this.value = MathHelper.clamp(this.value, 0.0F, 1.0F);
+				this.value = Mth.clamp(this.value, 0.0F, 1.0F);
 				float f = this.toValue(this.value);
 				this.clicked(f);
 				this.value = this.toPct(f);
 				this.getProxy().message = this.getMessage();
 			}
@@ -73,11 +73,11 @@
 	}
 
 	@Override
 	public void clicked(int i, int j) {
 		this.value = (float)(i - (this.getProxy().x + 4)) / (float)(this.getProxy().getWidth() - 8);
-		this.value = MathHelper.clamp(this.value, 0.0F, 1.0F);
+		this.value = Mth.clamp(this.value, 0.0F, 1.0F);
 		this.clicked(this.toValue(this.value));
 		this.getProxy().message = this.getMessage();
 		this.sliding = true;
 	}
 
