--- NetherCaveCarver.java
+++ NetherCaveCarver.java
@@ -1,11 +1,11 @@
 package net.minecraft.world.gen.carver;
 
 import java.util.Random;
 import net.minecraft.block.BlockState;
 import net.minecraft.block.Blocks;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.ChunkBlockStateStorage;
 
 public class NetherCaveCarver extends Carver {
 	protected void carveCave(long l, int i, int j, ChunkBlockStateStorage chunkBlockStateStorage, double d, double e, double f) {
@@ -32,17 +32,17 @@
 		}
 
 		int u = random.nextInt(n / 2) + n / 4;
 
 		for (boolean bl2 = random.nextInt(6) == 0; m < n; m++) {
-			double v = 1.5 + (double)(MathHelper.sin((float)m * (float) Math.PI / (float)n) * g * 1.0F);
+			double v = 1.5 + (double)(Mth.sin((float)m * (float) Math.PI / (float)n) * g * 1.0F);
 			double w = v * o;
-			float x = MathHelper.cos(k);
-			float y = MathHelper.sin(k);
-			d += (double)(MathHelper.cos(h) * x);
+			float x = Mth.cos(k);
+			float y = Mth.sin(k);
+			d += (double)(Mth.cos(h) * x);
 			e += (double)y;
-			f += (double)(MathHelper.sin(h) * x);
+			f += (double)(Mth.sin(h) * x);
 			if (bl2) {
 				k *= 0.92F;
 			} else {
 				k *= 0.7F;
 			}
@@ -67,16 +67,16 @@
 				if (z * z + aa * aa - ab * ab > ac * ac) {
 					return;
 				}
 
 				if (!(d < p - 16.0 - v * 2.0) && !(f < q - 16.0 - v * 2.0) && !(d > p + 16.0 + v * 2.0) && !(f > q + 16.0 + v * 2.0)) {
-					int ad = MathHelper.floor(d - v) - i * 16 - 1;
-					int ae = MathHelper.floor(d + v) - i * 16 + 1;
-					int af = MathHelper.floor(e - w) - 1;
-					int ag = MathHelper.floor(e + w) + 1;
-					int ah = MathHelper.floor(f - v) - j * 16 - 1;
-					int ai = MathHelper.floor(f + v) - j * 16 + 1;
+					int ad = Mth.floor(d - v) - i * 16 - 1;
+					int ae = Mth.floor(d + v) - i * 16 + 1;
+					int af = Mth.floor(e - w) - 1;
+					int ag = Mth.floor(e + w) + 1;
+					int ah = Mth.floor(f - v) - j * 16 - 1;
+					int ai = Mth.floor(f + v) - j * 16 + 1;
 					if (ad < 0) {
 						ad = 0;
 					}
 
 					if (ae > 16) {
