--- Village.java
+++ Village.java
@@ -17,11 +17,11 @@
 import net.minecraft.nbt.NbtList;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.UserCache;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
 
 public class Village {
 	private World world;
@@ -339,11 +339,11 @@
 		return integer != null ? integer : 0;
 	}
 
 	public int method_4505(String string, int i) {
 		int j = this.method_4504(string);
-		int k = MathHelper.clamp(j + i, -30, 10);
+		int k = Mth.clamp(j + i, -30, 10);
 		this.players.put(string, k);
 		return k;
 	}
 
 	public boolean method_4510(String string) {
