--- BakedQuadFactory.java
+++ BakedQuadFactory.java
@@ -286,19 +286,19 @@
 		int j = 7 * i;
 		float f = Float.intBitsToFloat(is[j]);
 		float g = Float.intBitsToFloat(is[j + 1]);
 		float h = Float.intBitsToFloat(is[j + 2]);
 		if (f < -0.1F || f >= 1.1F) {
-			f -= (float)MathHelper.floor(f);
+			f -= (float) MathHelper.floor(f);
 		}
 
 		if (g < -0.1F || g >= 1.1F) {
-			g -= (float)MathHelper.floor(g);
+			g -= (float) MathHelper.floor(g);
 		}
 
 		if (h < -0.1F || h >= 1.1F) {
-			h -= (float)MathHelper.floor(h);
+			h -= (float) MathHelper.floor(h);
 		}
 
 		float k = 0.0F;
 		float l = 0.0F;
 		switch (direction) {
