--- ModelLoader.java
+++ ModelLoader.java
@@ -688,11 +688,11 @@
 					ModelLoader.this.sprites.put(identifier, sprite);
 				}
 			}
 		};
 		this.atlas.method_10315(this.resourceManager, textureCreator);
-		this.sprites.put(new Identifier("missingno"), this.atlas.getTexture());
+		this.sprites.put(new Identifier("missingno"), this.atlas.getSprite());
 	}
 
 	private Set<Identifier> method_10410() {
 		Set<Identifier> set = Sets.newHashSet();
 
