--- CaveSpiderEntityRenderer.java
+++ CaveSpiderEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GlStateManager;
 import net.minecraft.entity.mob.CaveSpiderEntity;
 import net.minecraft.util.Identifier;
 
 public class CaveSpiderEntityRenderer extends SpiderEntityRenderer<CaveSpiderEntity> {
 	private static final Identifier TEXTURE = new Identifier("textures/entity/spider/cave_spider.png");
