--- EnderCrystalEntityRenderer.java
+++ EnderCrystalEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.entity.model.EnderCrystalEntityModel;
 import net.minecraft.client.render.entity.model.EntityModel;
 import net.minecraft.entity.EndCrystalEntity;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.math.MathHelper;
