--- FishingBobberEntityRenderer.java
+++ FishingBobberEntityRenderer.java
@@ -62,12 +62,12 @@
 			double w = (double)fishingBobberEntity.thrower.getEyeHeight();
 			if (this.dispatcher.options != null && this.dispatcher.options.perspective > 0 || fishingBobberEntity.thrower != MinecraftClient.getInstance().player) {
 				float x = (fishingBobberEntity.thrower.prevBodyYaw + (fishingBobberEntity.thrower.bodyYaw - fishingBobberEntity.thrower.prevBodyYaw) * h)
 					* (float) Math.PI
 					/ 180.0F;
-				double y = (double)MathHelper.sin(x);
-				double z = (double)MathHelper.cos(x);
+				double y = (double) MathHelper.sin(x);
+				double z = (double) MathHelper.cos(x);
 				double aa = 0.35;
 				double ab = 0.8;
 				t = fishingBobberEntity.thrower.prevX + (fishingBobberEntity.thrower.x - fishingBobberEntity.thrower.prevX) * (double)h - z * 0.35 - y * 0.8;
 				u = fishingBobberEntity.thrower.prevY + w + (fishingBobberEntity.thrower.y - fishingBobberEntity.thrower.prevY) * (double)h - 0.45;
 				v = fishingBobberEntity.thrower.prevZ + (fishingBobberEntity.thrower.z - fishingBobberEntity.thrower.prevZ) * (double)h - y * 0.35 + z * 0.8;
