--- HorseBaseEntityRenderer.java
+++ HorseBaseEntityRenderer.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.render.entity;
 
 import com.google.common.collect.Maps;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GlStateManager;
 import java.util.Map;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.render.entity.model.HorseBaseEntityModel;
 import net.minecraft.client.texture.LayeredTexture;
 import net.minecraft.entity.passive.HorseBaseEntity;
