--- ItemFrameEntityRenderer.java
+++ ItemFrameEntityRenderer.java
@@ -116,11 +116,11 @@
 						compassSprite.field_2151 = 0.0;
 						compassSprite.method_5241(
 							itemFrameEntity.world,
 							itemFrameEntity.x,
 							itemFrameEntity.z,
-							(double)MathHelper.wrapDegrees((float)(180 + itemFrameEntity.direction.getHorizontal() * 90)),
+							(double) MathHelper.wrapDegrees((float)(180 + itemFrameEntity.direction.getHorizontal() * 90)),
 							false,
 							true
 						);
 						compassSprite.field_2150 = d;
 						compassSprite.field_2151 = e;
