--- LeashKnotEntityRenderer.java
+++ LeashKnotEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.entity.model.LeashEntityModel;
 import net.minecraft.entity.decoration.LeashKnotEntity;
 import net.minecraft.util.Identifier;
 
 public class LeashKnotEntityRenderer extends EntityRenderer<LeashKnotEntity> {
