--- LivingEntityRenderer.java
+++ LivingEntityRenderer.java
@@ -37,11 +37,11 @@
 		super(entityRenderDispatcher);
 		this.model = entityModel;
 		this.shadowSize = f;
 	}
 
-	protected <V extends LivingEntity, U extends FeatureRenderer<V>> boolean addFeature(U featureRenderer) {
+	protected <V extends LivingEntity, U extends FeatureRenderer<V>> boolean addFeature(FeatureRenderer featureRenderer) {
 		return this.features.add(featureRenderer);
 	}
 
 	protected <V extends LivingEntity, U extends FeatureRenderer<V>> boolean removeFeature(U featureRenderer) {
 		return this.features.remove(featureRenderer);
@@ -269,11 +269,11 @@
 				this.buffer.put(l);
 				this.buffer.put(1.0F - h);
 			}
 
 			this.buffer.flip();
-			GL11.glTexEnv(8960, 8705, this.buffer);
+			GL11.glTexEnvfv(8960, 8705, this.buffer);
 			GlStateManager.activeTexture(GLX.texture);
 			GlStateManager.enableTexture();
 			GlStateManager.bindTexture(TEX.getGlId());
 			GL11.glTexEnvi(8960, 8704, GLX.combine);
 			GL11.glTexEnvi(8960, GLX.combineRgb, 8448);
