--- SkeletonEntityRenderer.java
+++ SkeletonEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.entity.feature.ArmorRenderer;
 import net.minecraft.client.render.entity.feature.HeldItemRenderer;
 import net.minecraft.client.render.entity.model.SkeletonEntityModel;
 import net.minecraft.entity.mob.SkeletonEntity;
 import net.minecraft.util.Identifier;
