--- VillagerEntityRenderer.java
+++ VillagerEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GlStateManager;
 import net.minecraft.client.render.entity.feature.HeadFeatureRenderer;
 import net.minecraft.client.render.entity.model.VillagerEntityModel;
 import net.minecraft.entity.passive.VillagerEntity;
 import net.minecraft.util.Identifier;
 
