--- BlockRenderManager.java
+++ BlockRenderManager.java
@@ -14,11 +14,11 @@
 import net.minecraft.resource.ResourceReloadListener;
 import net.minecraft.util.crash.CrashException;
 import net.minecraft.util.crash.CrashReport;
 import net.minecraft.util.crash.CrashReportSection;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.BlockView;
 import net.minecraft.world.level.LevelGeneratorType;
 
 public class BlockRenderManager implements ResourceReloadListener {
 	private BlockModelShapes models;
@@ -78,11 +78,11 @@
 	}
 
 	private BakedModel getModel(BlockState blockState, BlockPos blockPos) {
 		BakedModel bakedModel = this.models.getBakedModel(blockState);
 		if (blockPos != null && this.options.alternativeBlocks && bakedModel instanceof WeightedBakedModel) {
-			bakedModel = ((WeightedBakedModel)bakedModel).method_10425(MathHelper.hashCode(blockPos));
+			bakedModel = ((WeightedBakedModel)bakedModel).method_10425(Mth.hashCode(blockPos));
 		}
 
 		return bakedModel;
 	}
 
@@ -95,11 +95,11 @@
 			}
 		}
 
 		BakedModel bakedModel = this.models.getBakedModel(blockState);
 		if (blockPos != null && this.options.alternativeBlocks && bakedModel instanceof WeightedBakedModel) {
-			bakedModel = ((WeightedBakedModel)bakedModel).method_10425(MathHelper.hashCode(blockPos));
+			bakedModel = ((WeightedBakedModel)bakedModel).method_10425(Mth.hashCode(blockPos));
 		}
 
 		return bakedModel;
 	}
 
