--- BlockModelRenderer.java
+++ BlockModelRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.block;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import java.util.BitSet;
 import java.util.List;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
 import net.minecraft.client.MinecraftClient;
@@ -16,11 +16,11 @@
 import net.minecraft.util.crash.CrashException;
 import net.minecraft.util.crash.CrashReport;
 import net.minecraft.util.crash.CrashReportSection;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3i;
 import net.minecraft.world.BlockView;
 
 public class BlockModelRenderer {
 	public boolean render(BlockView blockView, BakedModel bakedModel, BlockState blockState, BlockPos blockPos, BufferBuilder bufferBuilder) {
@@ -110,11 +110,11 @@
 		double d = (double)blockPos.getX();
 		double e = (double)blockPos.getY();
 		double f = (double)blockPos.getZ();
 		Block.OffsetType offsetType = block.getOffsetType();
 		if (offsetType != Block.OffsetType.NONE) {
-			long l = MathHelper.hashCode(blockPos);
+			long l = Mth.hashCode(blockPos);
 			d += ((double)((float)(l >> 16 & 15L) / 15.0F) - 0.5) * 0.5;
 			f += ((double)((float)(l >> 24 & 15L) / 15.0F) - 0.5) * 0.5;
 			if (offsetType == Block.OffsetType.XYZ) {
 				e += ((double)((float)(l >> 20 & 15L) / 15.0F) - 1.0) * 0.2;
 			}
