--- MultiplayerScreen.java
+++ MultiplayerScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen.multiplayer;
 
+import com.element.render.MatrixStack;
 import com.google.common.base.Splitter;
 import com.google.common.collect.Lists;
 import java.util.List;
 import net.minecraft.client.gui.screen.AddServerScreen;
 import net.minecraft.client.gui.screen.ConfirmScreen;
@@ -259,16 +260,16 @@
 			}
 		}
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.tooltipText = null;
 		this.renderBackground();
 		this.serverListWidget.render(i, j, f);
 		this.drawCenteredString(this.textRenderer, I18n.translate("multiplayer.title"), this.width / 2, 20, 16777215);
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 		if (this.tooltipText != null) {
 			this.renderTooltip(Lists.newArrayList(Splitter.on("\n").split(this.tooltipText)), i, j);
 		}
 	}
 
