--- CapeFeatureRenderer.java
+++ CapeFeatureRenderer.java
@@ -30,11 +30,11 @@
 				- (abstractClientPlayerEntity.prevY + (abstractClientPlayerEntity.y - abstractClientPlayerEntity.prevY) * (double)h);
 			double m = abstractClientPlayerEntity.capeZ
 				+ (abstractClientPlayerEntity.prevCapeZ - abstractClientPlayerEntity.capeZ) * (double)h
 				- (abstractClientPlayerEntity.prevZ + (abstractClientPlayerEntity.z - abstractClientPlayerEntity.prevZ) * (double)h);
 			float n = abstractClientPlayerEntity.prevBodyYaw + (abstractClientPlayerEntity.bodyYaw - abstractClientPlayerEntity.prevBodyYaw) * h;
-			double o = (double)MathHelper.sin(n * (float) Math.PI / 180.0F);
+			double o = (double) MathHelper.sin(n * (float) Math.PI / 180.0F);
 			double p = (double)(-MathHelper.cos(n * (float) Math.PI / 180.0F));
 			float q = (float)e * 10.0F;
 			q = MathHelper.clamp(q, -6.0F, 32.0F);
 			float r = (float)(d * o + m * p) * 100.0F;
 			float s = (float)(d * p - m * o) * 100.0F;
