--- CapeFeatureRenderer.java
+++ CapeFeatureRenderer.java
@@ -1,12 +1,12 @@
 package net.minecraft.client.render.entity.feature;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.network.AbstractClientPlayerEntity;
 import net.minecraft.client.render.entity.PlayerEntityRenderer;
 import net.minecraft.client.render.entity.PlayerModelPart;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class CapeFeatureRenderer implements FeatureRenderer<AbstractClientPlayerEntity> {
 	private final PlayerEntityRenderer playerRenderer;
 
 	public CapeFeatureRenderer(PlayerEntityRenderer playerEntityRenderer) {
@@ -30,22 +30,22 @@
 				- (abstractClientPlayerEntity.prevY + (abstractClientPlayerEntity.y - abstractClientPlayerEntity.prevY) * (double)h);
 			double m = abstractClientPlayerEntity.capeZ
 				+ (abstractClientPlayerEntity.prevCapeZ - abstractClientPlayerEntity.capeZ) * (double)h
 				- (abstractClientPlayerEntity.prevZ + (abstractClientPlayerEntity.z - abstractClientPlayerEntity.prevZ) * (double)h);
 			float n = abstractClientPlayerEntity.prevBodyYaw + (abstractClientPlayerEntity.bodyYaw - abstractClientPlayerEntity.prevBodyYaw) * h;
-			double o = (double)MathHelper.sin(n * (float) Math.PI / 180.0F);
-			double p = (double)(-MathHelper.cos(n * (float) Math.PI / 180.0F));
+			double o = (double) Mth.sin(n * (float) Math.PI / 180.0F);
+			double p = (double)(-Mth.cos(n * (float) Math.PI / 180.0F));
 			float q = (float)e * 10.0F;
-			q = MathHelper.clamp(q, -6.0F, 32.0F);
+			q = Mth.clamp(q, -6.0F, 32.0F);
 			float r = (float)(d * o + m * p) * 100.0F;
 			float s = (float)(d * p - m * o) * 100.0F;
 			if (r < 0.0F) {
 				r = 0.0F;
 			}
 
 			float t = abstractClientPlayerEntity.prevStrideDistance + (abstractClientPlayerEntity.strideDistance - abstractClientPlayerEntity.prevStrideDistance) * h;
-			q += MathHelper.sin(
+			q += Mth.sin(
 					(abstractClientPlayerEntity.prevHorizontalSpeed + (abstractClientPlayerEntity.horizontalSpeed - abstractClientPlayerEntity.prevHorizontalSpeed) * h)
 						* 6.0F
 				)
 				* 32.0F
 				* t;
