--- BufferRenderer.java
+++ BufferRenderer.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.render;
 
-import com.mojang.blaze3d.platform.GLX;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GLX;
+import com.mojang.blaze3d.platform.GlStateManager;
 import java.nio.ByteBuffer;
 import java.util.List;
 import org.lwjgl.opengl.GL11;
 
 public class BufferRenderer {
