--- ClientTickTracker.java
+++ ClientTickTracker.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.render;
 
 import net.minecraft.client.MinecraftClient;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class ClientTickTracker {
 	float ticksPerSecond;
 	private double field_1041;
 	public int ticksThisFrame;
@@ -44,11 +44,11 @@
 		}
 
 		this.field_1042 = l;
 		double f = (d - this.field_1041) * this.field_1045;
 		this.field_1041 = d;
-		f = MathHelper.clamp(f, 0.0, 1.0);
+		f = Mth.clamp(f, 0.0, 1.0);
 		this.lastFrameDuration = (float)((double)this.lastFrameDuration + f * (double)this.field_1039 * (double)this.ticksPerSecond);
 		this.ticksThisFrame = (int)this.lastFrameDuration;
 		this.lastFrameDuration = this.lastFrameDuration - (float)this.ticksThisFrame;
 		if (this.ticksThisFrame > 10) {
 			this.ticksThisFrame = 10;
