--- Frustum.java
+++ Frustum.java
@@ -1,11 +1,11 @@
 package net.minecraft.client.render;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import java.nio.FloatBuffer;
 import net.minecraft.client.util.GlAllocationUtils;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class Frustum extends BaseFrustum {
 	private static Frustum instance = new Frustum();
 	private FloatBuffer projectionBuffer = GlAllocationUtils.allocateFloatBuffer(16);
 	private FloatBuffer modelBuffer = GlAllocationUtils.allocateFloatBuffer(16);
@@ -15,11 +15,11 @@
 		instance.start();
 		return instance;
 	}
 
 	private void normalize(float[] fs) {
-		float f = MathHelper.sqrt(fs[0] * fs[0] + fs[1] * fs[1] + fs[2] * fs[2]);
+		float f = Mth.sqrt(fs[0] * fs[0] + fs[1] * fs[1] + fs[2] * fs[2]);
 		fs[0] /= f;
 		fs[1] /= f;
 		fs[2] /= f;
 		fs[3] /= f;
 	}
