--- RealmsScreenProxy.java
+++ RealmsScreenProxy.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.realms;
 
+import com.element.render.MatrixStack;
 import com.google.common.collect.Lists;
 import java.util.Collections;
 import java.util.List;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.ButtonWidget;
@@ -65,11 +66,11 @@
 	public void renderBackground(int i) {
 		super.renderBackground(i);
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.realmsScreen.render(i, j, f);
 	}
 
 	@Override
 	public void renderTooltip(ItemStack itemStack, int i, int j) {
