--- MapState.java
+++ MapState.java
@@ -42,11 +42,11 @@
 	public void fromNbt(NbtCompound nbtCompound) {
 		this.dimensionId = nbtCompound.getByte("dimension");
 		this.xCenter = nbtCompound.getInt("xCenter");
 		this.zCenter = nbtCompound.getInt("zCenter");
 		this.scale = nbtCompound.getByte("scale");
-		this.scale = (byte)MathHelper.clamp(this.scale, 0, 4);
+		this.scale = (byte) MathHelper.clamp(this.scale, 0, 4);
 		int i = nbtCompound.getShort("width");
 		int j = nbtCompound.getShort("height");
 		if (i == 128 && j == 128) {
 			this.colors = nbtCompound.getByteArray("colors");
 		} else {
