--- BannerItem.java
+++ BannerItem.java
@@ -12,11 +12,11 @@
 import net.minecraft.nbt.NbtList;
 import net.minecraft.util.CommonI18n;
 import net.minecraft.util.DyeColor;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class BannerItem extends BlockItem {
 	public BannerItem() {
 		super(Blocks.STANDING_BANNER);
@@ -40,11 +40,11 @@
 				return false;
 			} else if (world.isClient) {
 				return true;
 			} else {
 				if (direction == Direction.UP) {
-					int i = MathHelper.floor((double)((playerEntity.yaw + 180.0F) * 16.0F / 360.0F) + 0.5) & 15;
+					int i = Mth.floor((double)((playerEntity.yaw + 180.0F) * 16.0F / 360.0F) + 0.5) & 15;
 					world.setBlockState(blockPos, Blocks.STANDING_BANNER.getDefaultState().with(StandingSignBlock.ROTATION, i), 3);
 				} else {
 					world.setBlockState(blockPos, Blocks.WALL_BANNER.getDefaultState().with(WallSignBlock.FACING, direction), 3);
 				}
 
