--- BoatItem.java
+++ BoatItem.java
@@ -8,11 +8,11 @@
 import net.minecraft.item.itemgroup.ItemGroup;
 import net.minecraft.stat.Stats;
 import net.minecraft.util.hit.BlockHitResult;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
 
 public class BoatItem extends Item {
 	public BoatItem() {
@@ -27,14 +27,14 @@
 		float h = playerEntity.prevYaw + (playerEntity.yaw - playerEntity.prevYaw) * f;
 		double d = playerEntity.prevX + (playerEntity.x - playerEntity.prevX) * (double)f;
 		double e = playerEntity.prevY + (playerEntity.y - playerEntity.prevY) * (double)f + (double)playerEntity.getEyeHeight();
 		double i = playerEntity.prevZ + (playerEntity.z - playerEntity.prevZ) * (double)f;
 		Vec3d vec3d = new Vec3d(d, e, i);
-		float j = MathHelper.cos(-h * (float) (Math.PI / 180.0) - (float) Math.PI);
-		float k = MathHelper.sin(-h * (float) (Math.PI / 180.0) - (float) Math.PI);
-		float l = -MathHelper.cos(-g * (float) (Math.PI / 180.0));
-		float m = MathHelper.sin(-g * (float) (Math.PI / 180.0));
+		float j = Mth.cos(-h * (float) (Math.PI / 180.0) - (float) Math.PI);
+		float k = Mth.sin(-h * (float) (Math.PI / 180.0) - (float) Math.PI);
+		float l = -Mth.cos(-g * (float) (Math.PI / 180.0));
+		float m = Mth.sin(-g * (float) (Math.PI / 180.0));
 		float n = k * l;
 		float p = j * l;
 		double q = 5.0;
 		Vec3d vec3d2 = vec3d.add((double)n * q, (double)m * q, (double)p * q);
 		BlockHitResult blockHitResult = world.rayTrace(vec3d, vec3d2, true);
@@ -69,11 +69,11 @@
 					}
 
 					BoatEntity boatEntity = new BoatEntity(
 						world, (double)((float)blockPos.getX() + 0.5F), (double)((float)blockPos.getY() + 1.0F), (double)((float)blockPos.getZ() + 0.5F)
 					);
-					boatEntity.yaw = (float)(((MathHelper.floor((double)(playerEntity.yaw * 4.0F / 360.0F) + 0.5) & 3) - 1) * 90);
+					boatEntity.yaw = (float)(((Mth.floor((double)(playerEntity.yaw * 4.0F / 360.0F) + 0.5) & 3) - 1) * 90);
 					if (!world.doesBoxCollide(boatEntity, boatEntity.getBoundingBox().expand(-0.1, -0.1, -0.1)).isEmpty()) {
 						return itemStack;
 					}
 
 					if (!world.isClient) {
