--- FilledMapItem.java
+++ FilledMapItem.java
@@ -14,11 +14,11 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.map.MapState;
 import net.minecraft.network.Packet;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.Chunk;
 
 public class FilledMapItem extends NetworkSyncedItem {
 	protected FilledMapItem() {
@@ -56,12 +56,12 @@
 	public void updateColors(World world, Entity entity, MapState mapState) {
 		if (world.dimension.getType() == mapState.dimensionId && entity instanceof PlayerEntity) {
 			int i = 1 << mapState.scale;
 			int j = mapState.xCenter;
 			int k = mapState.zCenter;
-			int l = MathHelper.floor(entity.x - (double)j) / i + 64;
-			int m = MathHelper.floor(entity.z - (double)k) / i + 64;
+			int l = Mth.floor(entity.x - (double)j) / i + 64;
+			int m = Mth.floor(entity.z - (double)k) / i + 64;
 			int n = 128 / i;
 			if (world.dimension.hasNoSkylight()) {
 				n /= 2;
 			}
 
