--- SignItem.java
+++ SignItem.java
@@ -7,11 +7,11 @@
 import net.minecraft.block.entity.SignBlockEntity;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.itemgroup.ItemGroup;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class SignItem extends Item {
 	public SignItem() {
 		this.maxCount = 16;
@@ -32,11 +32,11 @@
 				return false;
 			} else if (world.isClient) {
 				return true;
 			} else {
 				if (direction == Direction.UP) {
-					int i = MathHelper.floor((double)((playerEntity.yaw + 180.0F) * 16.0F / 360.0F) + 0.5) & 15;
+					int i = Mth.floor((double)((playerEntity.yaw + 180.0F) * 16.0F / 360.0F) + 0.5) & 15;
 					world.setBlockState(blockPos, Blocks.STANDING_SIGN.getDefaultState().with(StandingSignBlock.ROTATION, i), 3);
 				} else {
 					world.setBlockState(blockPos, Blocks.WALL_SIGN.getDefaultState().with(WallSignBlock.FACING, direction), 3);
 				}
 
