--- SkullItem.java
+++ SkullItem.java
@@ -13,11 +13,11 @@
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.nbt.NbtHelper;
 import net.minecraft.util.CommonI18n;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class SkullItem extends Item {
 	private static final String[] SKULL_TYPES = new String[]{"skeleton", "wither", "zombie", "char", "creeper"};
 
@@ -50,11 +50,11 @@
 			} else {
 				if (!world.isClient) {
 					world.setBlockState(blockPos, Blocks.SKULL.getDefaultState().with(SkullBlock.FACING, direction), 3);
 					int i = 0;
 					if (direction == Direction.UP) {
-						i = MathHelper.floor((double)(playerEntity.yaw * 16.0F / 360.0F) + 0.5) & 15;
+						i = Mth.floor((double)(playerEntity.yaw * 16.0F / 360.0F) + 0.5) & 15;
 					}
 
 					BlockEntity blockEntity = world.getBlockEntity(blockPos);
 					if (blockEntity instanceof SkullBlockEntity) {
 						SkullBlockEntity skullBlockEntity = (SkullBlockEntity)blockEntity;
