--- SpawnEggItem.java
+++ SpawnEggItem.java
@@ -17,11 +17,11 @@
 import net.minecraft.stat.Stats;
 import net.minecraft.util.CommonI18n;
 import net.minecraft.util.hit.BlockHitResult;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class SpawnEggItem extends Item {
 	public SpawnEggItem() {
 		this.setUnbreakable(true);
@@ -141,11 +141,11 @@
 
 			for (int j = 0; j < 1; j++) {
 				entity = EntityType.createInstanceFromRawId(i, world);
 				if (entity instanceof LivingEntity) {
 					MobEntity mobEntity = (MobEntity)entity;
-					entity.refreshPositionAndAngles(d, e, f, MathHelper.wrapDegrees(world.random.nextFloat() * 360.0F), 0.0F);
+					entity.refreshPositionAndAngles(d, e, f, Mth.wrapDegrees(world.random.nextFloat() * 360.0F), 0.0F);
 					mobEntity.headYaw = mobEntity.yaw;
 					mobEntity.bodyYaw = mobEntity.yaw;
 					mobEntity.initialize(world.getLocalDifficulty(new BlockPos(mobEntity)), null);
 					world.spawnEntity(entity);
 					mobEntity.playAmbientSound();
