--- FireworksSparkParticle.java
+++ FireworksSparkParticle.java
@@ -295,11 +295,11 @@
 				for (int k = -i; k <= i; k++) {
 					for (int l = -i; l <= i; l++) {
 						double h = (double)k + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
 						double m = (double)j + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
 						double n = (double)l + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
-						double o = (double)MathHelper.sqrt(h * h + m * m + n * n) / d + this.random.nextGaussian() * 0.05;
+						double o = (double) MathHelper.sqrt(h * h + m * m + n * n) / d + this.random.nextGaussian() * 0.05;
 						this.addExplosionParticle(e, f, g, h / o, m / o, n / o, is, js, bl, bl2);
 						if (j != -i && j != i && k != -i && k != i) {
 							l += i * 2 - 1;
 						}
 					}
