--- RainSplashParticle.java
+++ RainSplashParticle.java
@@ -57,11 +57,11 @@
 				d = (double)(1.0F - AbstractFluidBlock.getHeightPercent((Integer)blockState.get(AbstractFluidBlock.LEVEL)));
 			} else {
 				d = block.getMaxY();
 			}
 
-			double e = (double)MathHelper.floor(this.y) + d;
+			double e = (double) MathHelper.floor(this.y) + d;
 			if (this.y < e) {
 				this.remove();
 			}
 		}
 	}
