--- ResourcePack.java
+++ ResourcePack.java
@@ -1,10 +1,11 @@
 package net.minecraft.resource;
 
 import java.awt.image.BufferedImage;
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.List;
 import java.util.Set;
 import net.minecraft.client.resource.ResourceMetadataProvider;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.MetadataSerializer;
 
@@ -17,7 +18,9 @@
 
 	<T extends ResourceMetadataProvider> T parseMetadata(MetadataSerializer var1, String var2) throws IOException;
 
 	BufferedImage getIcon() throws IOException;
 
+	List<Identifier> getAllResources() throws IOException;
+
 	String getName();
 }
