--- EnderDragonEntity.java
+++ EnderDragonEntity.java
@@ -177,11 +177,11 @@
 
 					if (this.field_3744 || k < 100.0 || k > 22500.0 || this.horizontalCollision || this.verticalCollision) {
 						this.method_2906();
 					}
 
-					e /= (double)MathHelper.sqrt(d * d + j * j);
+					e /= (double) MathHelper.sqrt(d * d + j * j);
 					float p = 0.6F;
 					e = MathHelper.clamp(e, (double)(-p), (double)p);
 					this.velocityY += e * 0.1F;
 					this.yaw = MathHelper.wrapDegrees(this.yaw);
 					double q = 180.0 - MathHelper.atan2(d, j) * 180.0 / (float) Math.PI;
@@ -194,11 +194,11 @@
 						r = -50.0;
 					}
 
 					Vec3d vec3d = new Vec3d(this.field_3742 - this.x, this.field_3751 - this.y, this.field_3752 - this.z).normalize();
 					double o = (double)(-MathHelper.cos(this.yaw * (float) Math.PI / 180.0F));
-					Vec3d vec3d2 = new Vec3d((double)MathHelper.sin(this.yaw * (float) Math.PI / 180.0F), this.velocityY, o).normalize();
+					Vec3d vec3d2 = new Vec3d((double) MathHelper.sin(this.yaw * (float) Math.PI / 180.0F), this.velocityY, o).normalize();
 					float s = ((float)vec3d2.dotProduct(vec3d) + 0.5F) / 1.5F;
 					if (s < 0.0F) {
 						s = 0.0F;
 					}
 
@@ -389,11 +389,11 @@
 			this.target = null;
 		}
 	}
 
 	private float wrapYawChange(double d) {
-		return (float)MathHelper.wrapDegrees(d);
+		return (float) MathHelper.wrapDegrees(d);
 	}
 
 	private boolean destroyBlocks(Box box) {
 		int i = MathHelper.floor(box.minX);
 		int j = MathHelper.floor(box.minY);
