--- EyeOfEnderEntity.java
+++ EyeOfEnderEntity.java
@@ -110,11 +110,11 @@
 		this.yaw = this.prevYaw + (this.yaw - this.prevYaw) * 0.2F;
 		if (!this.world.isClient) {
 			double d = this.targetX - this.x;
 			double e = this.targetZ - this.z;
 			float g = (float)Math.sqrt(d * d + e * e);
-			float h = (float)MathHelper.atan2(e, d);
+			float h = (float) MathHelper.atan2(e, d);
 			double i = (double)f + (double)(g - f) * 0.0025;
 			if (g < 1.0F) {
 				i *= 0.8;
 				this.velocityY *= 0.8;
 			}
