--- Biome.java
+++ Biome.java
@@ -420,18 +420,18 @@
 	public void decorate(World world, Random random, BlockPos blockPos) {
 		this.biomeDecorator.decorate(world, random, this, blockPos);
 	}
 
 	public int getGrassColor(BlockPos blockPos) {
-		double d = (double)MathHelper.clamp(this.getTemperature(blockPos), 0.0F, 1.0F);
-		double e = (double)MathHelper.clamp(this.getRainfall(), 0.0F, 1.0F);
+		double d = (double) MathHelper.clamp(this.getTemperature(blockPos), 0.0F, 1.0F);
+		double e = (double) MathHelper.clamp(this.getRainfall(), 0.0F, 1.0F);
 		return GrassColors.getColor(d, e);
 	}
 
 	public int getFoliageColor(BlockPos blockPos) {
-		double d = (double)MathHelper.clamp(this.getTemperature(blockPos), 0.0F, 1.0F);
-		double e = (double)MathHelper.clamp(this.getRainfall(), 0.0F, 1.0F);
+		double d = (double) MathHelper.clamp(this.getTemperature(blockPos), 0.0F, 1.0F);
+		double e = (double) MathHelper.clamp(this.getRainfall(), 0.0F, 1.0F);
 		return FoliageColors.getColor(d, e);
 	}
 
 	public boolean isMutatedBiome() {
 		return this.mutated;
