--- Explosion.java
+++ Explosion.java
@@ -117,11 +117,11 @@
 				double x = entity.distanceTo(this.x, this.y, this.z) / (double)r;
 				if (x <= 1.0) {
 					double y = entity.x - this.x;
 					double z = entity.y + (double)entity.getEyeHeight() - this.y;
 					double aa = entity.z - this.z;
-					double ab = (double)MathHelper.sqrt(y * y + z * z + aa * aa);
+					double ab = (double) MathHelper.sqrt(y * y + z * z + aa * aa);
 					if (ab != 0.0) {
 						y /= ab;
 						z /= ab;
 						aa /= ab;
 						double ac = (double)this.world.method_3612(vec3d, entity.getBoundingBox());
@@ -156,11 +156,11 @@
 					double e = (double)((float)blockPos.getY() + this.world.random.nextFloat());
 					double f = (double)((float)blockPos.getZ() + this.world.random.nextFloat());
 					double g = d - this.x;
 					double h = e - this.y;
 					double i = f - this.z;
-					double j = (double)MathHelper.sqrt(g * g + h * h + i * i);
+					double j = (double) MathHelper.sqrt(g * g + h * h + i * i);
 					g /= j;
 					h /= j;
 					i /= j;
 					double k = 0.5 / (j / (double)this.power + 0.1);
 					k *= (double)(this.world.random.nextFloat() * this.world.random.nextFloat() + 0.3F);
