--- ConfirmScreen.java
+++ ConfirmScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen;
 
+import com.element.render.MatrixStack;
 import com.google.common.collect.Lists;
 import java.util.List;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.gui.widget.IdentifiableBooleanConsumer;
 import net.minecraft.client.gui.widget.OptionButtonWidget;
@@ -47,21 +48,21 @@
 	protected void buttonClicked(ButtonWidget buttonWidget) {
 		this.consumer.confirmResult(buttonWidget.id == 0, this.identifier);
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		this.drawCenteredString(this.textRenderer, this.title, this.width / 2, 70, 16777215);
 		int k = 90;
 
 		for (String string : this.lines) {
 			this.drawCenteredString(this.textRenderer, string, this.width / 2, k, 16777215);
 			k += this.textRenderer.fontHeight;
 		}
 
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 
 	public void disableButtons(int i) {
 		this.buttonEnableTimer = i;
 
