--- ConnectScreen.java
+++ ConnectScreen.java
@@ -1,10 +1,12 @@
 package net.minecraft.client.gui.screen;
 
 import java.net.InetAddress;
 import java.net.UnknownHostException;
 import java.util.concurrent.atomic.AtomicInteger;
+
+import com.element.render.MatrixStack;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.network.ClientLoginNetworkHandler;
 import net.minecraft.client.network.ServerInfo;
 import net.minecraft.client.resource.language.I18n;
@@ -118,16 +120,16 @@
 			this.client.setScreen(this.parent);
 		}
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		if (this.connection == null) {
 			this.drawCenteredString(this.textRenderer, I18n.translate("connect.connecting"), this.width / 2, this.height / 2 - 50, 16777215);
 		} else {
 			this.drawCenteredString(this.textRenderer, I18n.translate("connect.authorizing"), this.width / 2, this.height / 2 - 50, 16777215);
 		}
 
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 }
