--- CreditsScreen.java
+++ CreditsScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen;
 
+import com.element.render.MatrixStack;
 import com.google.common.collect.Lists;
 import com.mojang.blaze3d.platform.GlStateManager;
 import java.io.BufferedReader;
 import java.io.InputStream;
 import java.io.InputStreamReader;
@@ -143,11 +144,11 @@
 		bufferBuilder.vertex(0.0, 0.0, (double)this.zOffset).texture(0.0, (double)(h * l)).color(m, m, m, 1.0F).next();
 		tessellator.draw();
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground(i, j, f);
 		Tessellator tessellator = Tessellator.getInstance();
 		BufferBuilder bufferBuilder = tessellator.getBuffer();
 		int k = 274;
 		int l = this.width / 2 - k / 2;
@@ -193,8 +194,8 @@
 		bufferBuilder.vertex((double)o, (double)p, (double)this.zOffset).texture(1.0, 1.0).color(1.0F, 1.0F, 1.0F, 1.0F).next();
 		bufferBuilder.vertex((double)o, 0.0, (double)this.zOffset).texture(1.0, 0.0).color(1.0F, 1.0F, 1.0F, 1.0F).next();
 		bufferBuilder.vertex(0.0, 0.0, (double)this.zOffset).texture(0.0, 0.0).color(1.0F, 1.0F, 1.0F, 1.0F).next();
 		tessellator.draw();
 		GlStateManager.disableBlend();
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 }
