--- CustomizedWorldPresetsScreen.java
+++ CustomizedWorldPresetsScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen;
 
+import com.element.render.MatrixStack;
 import com.google.common.collect.Lists;
 import com.mojang.blaze3d.platform.GlStateManager;
 import java.util.List;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.gui.widget.ListWidget;
@@ -80,18 +81,18 @@
 				this.client.setScreen(this.parent);
 		}
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		this.presetList.render(i, j, f);
 		this.drawCenteredString(this.textRenderer, this.title, this.width / 2, 8, 16777215);
 		this.drawWithShadow(this.textRenderer, this.shareText, 50, 30, 10526880);
 		this.drawWithShadow(this.textRenderer, this.listText, 50, 70, 10526880);
 		this.searchField.render();
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 
 	@Override
 	public void tick() {
 		this.searchField.tick();
