--- DemoScreen.java
+++ DemoScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen;
 
+import com.element.render.MatrixStack;
 import com.mojang.blaze3d.platform.GlStateManager;
 import java.net.URI;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.option.GameOptions;
 import net.minecraft.client.resource.language.I18n;
@@ -55,11 +56,11 @@
 		int j = (this.height - 166) / 2;
 		this.drawTexture(i, j, 0, 0, 248, 166);
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		int k = (this.width - 248) / 2 + 10;
 		int l = (this.height - 166) / 2 + 8;
 		this.textRenderer.draw(I18n.translate("demo.help.title"), k, l, 2039583);
 		l += 12;
@@ -79,8 +80,8 @@
 			);
 		this.textRenderer.draw(I18n.translate("demo.help.movementMouse"), k, l + 12, 5197647);
 		this.textRenderer.draw(I18n.translate("demo.help.jump", GameOptions.getFormattedNameForKeyCode(gameOptions.jumpKey.getCode())), k, l + 24, 5197647);
 		this.textRenderer.draw(I18n.translate("demo.help.inventory", GameOptions.getFormattedNameForKeyCode(gameOptions.inventoryKey.getCode())), k, l + 36, 5197647);
 		this.textRenderer.drawTrimmed(I18n.translate("demo.help.fullWrapped"), k, l + 68, 218, 2039583);
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 }
