--- OpenToLanScreen.java
+++ OpenToLanScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen;
 
+import com.element.render.MatrixStack;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.resource.language.I18n;
 import net.minecraft.text.LiteralText;
 import net.minecraft.text.Text;
 import net.minecraft.text.TranslatableText;
@@ -70,12 +71,12 @@
 			this.client.inGameHud.getChatHud().addMessage(text);
 		}
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		this.drawCenteredString(this.textRenderer, I18n.translate("lanServer.title"), this.width / 2, 50, 16777215);
 		this.drawCenteredString(this.textRenderer, I18n.translate("lanServer.otherPlayers"), this.width / 2, 82, 16777215);
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 }
