--- ResourcePackScreen.java
+++ ResourcePackScreen.java
@@ -1,8 +1,10 @@
 package net.minecraft.client.gui.screen;
 
 import com.google.common.collect.Lists;
+
+import java.awt.*;
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
 import java.util.Collections;
 import java.util.List;
@@ -123,11 +125,16 @@
 					bl = true;
 				}
 
 				if (bl) {
 					LOGGER.info("Opening via system class!");
-					Sys.openURL("file://" + string);
+					try {
+						Desktop.getDesktop().open(new File(string));
+					} catch (Throwable var6) {
+						LOGGER.error("Couldn't open link", var6);
+						return;
+					}
 				}
 			} else if (buttonWidget.id == 1) {
 				if (this.dirty) {
 					List<ResourcePackLoader.Entry> list = Lists.newArrayList();
 
