--- VideoOptionsScreen.java
+++ VideoOptionsScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen;
 
+import com.element.render.MatrixStack;
 import com.mojang.blaze3d.platform.GLX;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.gui.widget.EntryListWidget;
 import net.minecraft.client.gui.widget.OptionPairWidget;
 import net.minecraft.client.option.GameOptions;
@@ -102,12 +103,12 @@
 			this.init(this.client, m, n);
 		}
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		this.list.render(i, j, f);
 		this.drawCenteredString(this.textRenderer, this.title, this.width / 2, 5, 16777215);
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 }
