--- StrongholdStructure.java
+++ StrongholdStructure.java
@@ -5,11 +5,11 @@
 import java.util.Map;
 import java.util.Random;
 import java.util.Map.Entry;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.ChunkPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.gen.GeneratorConfig;
 
 public class StrongholdStructure extends StructureFeature {
@@ -32,15 +32,15 @@
 	public StrongholdStructure(Map<String, String> map) {
 		this();
 
 		for (Entry<String, String> entry : map.entrySet()) {
 			if (((String)entry.getKey()).equals("distance")) {
-				this.distance = MathHelper.parseDouble((String)entry.getValue(), this.distance, 1.0);
+				this.distance = Mth.parseDouble((String)entry.getValue(), this.distance, 1.0);
 			} else if (((String)entry.getKey()).equals("count")) {
-				this.count = new ChunkPos[MathHelper.parseInt((String)entry.getValue(), this.count.length, 1)];
+				this.count = new ChunkPos[Mth.parseInt((String)entry.getValue(), this.count.length, 1)];
 			} else if (((String)entry.getKey()).equals("spread")) {
-				this.spread = MathHelper.parseInt((String)entry.getValue(), this.spread, 1);
+				this.spread = Mth.parseInt((String)entry.getValue(), this.spread, 1);
 			}
 		}
 	}
 
 	@Override
