--- VillageStructure.java
+++ VillageStructure.java
@@ -4,11 +4,11 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Random;
 import java.util.Map.Entry;
 import net.minecraft.nbt.NbtCompound;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.gen.GeneratorConfig;
 
 public class VillageStructure extends StructureFeature {
@@ -23,13 +23,13 @@
 	public VillageStructure(Map<String, String> map) {
 		this();
 
 		for (Entry<String, String> entry : map.entrySet()) {
 			if (((String)entry.getKey()).equals("size")) {
-				this.size = MathHelper.parseInt((String)entry.getValue(), this.size, 0);
+				this.size = Mth.parseInt((String)entry.getValue(), this.size, 0);
 			} else if (((String)entry.getKey()).equals("distance")) {
-				this.distance = MathHelper.parseInt((String)entry.getValue(), this.distance, this.field_4973 + 1);
+				this.distance = Mth.parseInt((String)entry.getValue(), this.distance, this.field_4973 + 1);
 			}
 		}
 	}
 
 	@Override
