--- AbstractArrowEntity.java
+++ AbstractArrowEntity.java
@@ -62,11 +62,11 @@
 
 		this.y = livingEntity.y + (double)livingEntity.getEyeHeight() - 0.1F;
 		double d = livingEntity2.x - livingEntity.x;
 		double e = livingEntity2.getBoundingBox().minY + (double)(livingEntity2.height / 3.0F) - this.y;
 		double h = livingEntity2.z - livingEntity.z;
-		double i = (double)MathHelper.sqrt(d * d + h * h);
+		double i = (double) MathHelper.sqrt(d * d + h * h);
 		if (!(i < 1.0E-7)) {
 			float j = (float)(MathHelper.atan2(h, d) * 180.0 / (float) Math.PI) - 90.0F;
 			float k = (float)(-(MathHelper.atan2(e, i) * 180.0 / (float) Math.PI));
 			double l = d / i;
 			double m = h / i;
