--- AnvilScreen.java
+++ AnvilScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen.ingame;
 
+import com.element.render.MatrixStack;
 import com.mojang.blaze3d.platform.GlStateManager;
 import io.netty.buffer.Unpooled;
 import java.util.List;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.widget.TextFieldWidget;
@@ -116,12 +117,12 @@
 		super.mouseClicked(i, j, k);
 		this.renameTextField.mouseClicked(i, j, k);
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
-		super.render(i, j, f);
+	public void render(MatrixStack matrices, int j, float f, int i) {
+		super.render(matrices, j, f, i);
 		GlStateManager.disableLighting();
 		GlStateManager.disableBlend();
 		this.renameTextField.render();
 	}
 
