--- EnchantingScreen.java
+++ EnchantingScreen.java
@@ -107,12 +107,12 @@
 		GlStateManager.translate((1.0F - m) * 0.2F, (1.0F - m) * 0.1F, (1.0F - m) * 0.25F);
 		GlStateManager.rotate(-(1.0F - m) * 90.0F - 90.0F, 0.0F, 1.0F, 0.0F);
 		GlStateManager.rotate(180.0F, 1.0F, 0.0F, 0.0F);
 		float n = this.pageAngle + (this.nextPageAngle - this.pageAngle) * f + 0.25F;
 		float o = this.pageAngle + (this.nextPageAngle - this.pageAngle) * f + 0.75F;
-		n = (n - (float)MathHelper.fastFloor((double)n)) * 1.6F - 0.3F;
-		o = (o - (float)MathHelper.fastFloor((double)o)) * 1.6F - 0.3F;
+		n = (n - (float) MathHelper.fastFloor((double)n)) * 1.6F - 0.3F;
+		o = (o - (float) MathHelper.fastFloor((double)o)) * 1.6F - 0.3F;
 		if (n < 0.0F) {
 			n = 0.0F;
 		}
 
 		if (o < 0.0F) {
