--- HandledScreen.java
+++ HandledScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen.ingame;
 
+import com.element.render.MatrixStack;
 import com.google.common.collect.Sets;
 import com.mojang.blaze3d.platform.GLX;
 import com.mojang.blaze3d.platform.GlStateManager;
 import java.util.Set;
 import net.minecraft.client.MinecraftClient;
@@ -60,20 +61,20 @@
 		this.x = (this.width - this.backgroundWidth) / 2;
 		this.y = (this.height - this.backgroundHeight) / 2;
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		int k = this.x;
 		int l = this.y;
 		this.drawBackground(f, i, j);
 		GlStateManager.disableRescaleNormal();
 		DiffuseLighting.disable();
 		GlStateManager.disableLighting();
 		GlStateManager.disableDepthTest();
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 		DiffuseLighting.enable();
 		GlStateManager.pushMatrix();
 		GlStateManager.translate((float)k, (float)l, 0.0F);
 		GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
 		GlStateManager.enableRescaleNormal();
