--- BeaconBlockEntityRenderer.java
+++ BeaconBlockEntityRenderer.java
@@ -33,11 +33,11 @@
 				GlStateManager.disableCull();
 				GlStateManager.disableBlend();
 				GlStateManager.depthMask(true);
 				GlStateManager.blendFuncSeparate(770, 1, 1, 0);
 				double m = (double)beaconBlockEntity.getEntityWorld().getLastUpdateTime() + (double)g;
-				double n = MathHelper.fractionalPart(-m * 0.2 - (double)MathHelper.floor(-m * 0.1));
+				double n = MathHelper.fractionalPart(-m * 0.2 - (double) MathHelper.floor(-m * 0.1));
 				float o = beamSegment.getColor()[0];
 				float p = beamSegment.getColor()[1];
 				float q = beamSegment.getColor()[2];
 				double r = m * 0.025 * -1.5;
 				double s = 0.2;
