--- ChestBlockEntityRenderer.java
+++ ChestBlockEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.block.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import java.util.Calendar;
 import net.minecraft.block.Block;
 import net.minecraft.block.ChestBlock;
 import net.minecraft.block.entity.ChestBlockEntity;
 import net.minecraft.client.render.entity.model.ChestBlockEntityModel;
