--- EnchantingTableBlockEntityRenderer.java
+++ EnchantingTableBlockEntityRenderer.java
@@ -1,22 +1,22 @@
 package net.minecraft.client.render.block.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.block.entity.EnchantingTableBlockEntity;
 import net.minecraft.client.render.entity.model.BookModel;
 import net.minecraft.util.Identifier;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class EnchantingTableBlockEntityRenderer extends BlockEntityRenderer<EnchantingTableBlockEntity> {
 	private static final Identifier BOOK_TEXTURE = new Identifier("textures/entity/enchanting_table_book.png");
 	private BookModel bookModel = new BookModel();
 
 	public void render(EnchantingTableBlockEntity enchantingTableBlockEntity, double d, double e, double f, float g, int i) {
 		GlStateManager.pushMatrix();
 		GlStateManager.translate((float)d + 0.5F, (float)e + 0.75F, (float)f + 0.5F);
 		float h = (float)enchantingTableBlockEntity.ticks + g;
-		GlStateManager.translate(0.0F, 0.1F + MathHelper.sin(h * 0.1F) * 0.01F, 0.0F);
+		GlStateManager.translate(0.0F, 0.1F + Mth.sin(h * 0.1F) * 0.01F, 0.0F);
 		float j = enchantingTableBlockEntity.openBookAngle - enchantingTableBlockEntity.openBookAnglePrev;
 
 		while (j >= (float) Math.PI) {
 			j -= (float) (Math.PI * 2);
 		}
@@ -29,12 +29,12 @@
 		GlStateManager.rotate(-k * 180.0F / (float) Math.PI, 0.0F, 1.0F, 0.0F);
 		GlStateManager.rotate(80.0F, 0.0F, 0.0F, 1.0F);
 		this.bindTexture(BOOK_TEXTURE);
 		float l = enchantingTableBlockEntity.pageAngle + (enchantingTableBlockEntity.nextPageAngle - enchantingTableBlockEntity.pageAngle) * g + 0.25F;
 		float m = enchantingTableBlockEntity.pageAngle + (enchantingTableBlockEntity.nextPageAngle - enchantingTableBlockEntity.pageAngle) * g + 0.75F;
-		l = (l - (float)MathHelper.fastFloor((double)l)) * 1.6F - 0.3F;
-		m = (m - (float)MathHelper.fastFloor((double)m)) * 1.6F - 0.3F;
+		l = (l - (float) Mth.fastFloor((double)l)) * 1.6F - 0.3F;
+		m = (m - (float) Mth.fastFloor((double)m)) * 1.6F - 0.3F;
 		if (l < 0.0F) {
 			l = 0.0F;
 		}
 
 		if (m < 0.0F) {
