--- EnchantingTableBlockEntityRenderer.java
+++ EnchantingTableBlockEntityRenderer.java
@@ -29,12 +29,12 @@
 		GlStateManager.rotate(-k * 180.0F / (float) Math.PI, 0.0F, 1.0F, 0.0F);
 		GlStateManager.rotate(80.0F, 0.0F, 0.0F, 1.0F);
 		this.bindTexture(BOOK_TEXTURE);
 		float l = enchantingTableBlockEntity.pageAngle + (enchantingTableBlockEntity.nextPageAngle - enchantingTableBlockEntity.pageAngle) * g + 0.25F;
 		float m = enchantingTableBlockEntity.pageAngle + (enchantingTableBlockEntity.nextPageAngle - enchantingTableBlockEntity.pageAngle) * g + 0.75F;
-		l = (l - (float)MathHelper.fastFloor((double)l)) * 1.6F - 0.3F;
-		m = (m - (float)MathHelper.fastFloor((double)m)) * 1.6F - 0.3F;
+		l = (l - (float) MathHelper.fastFloor((double)l)) * 1.6F - 0.3F;
+		m = (m - (float) MathHelper.fastFloor((double)m)) * 1.6F - 0.3F;
 		if (l < 0.0F) {
 			l = 0.0F;
 		}
 
 		if (m < 0.0F) {
