--- EnderChestBlockEntityRenderer.java
+++ EnderChestBlockEntityRenderer.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.block.entity;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GlStateManager;
 import net.minecraft.block.entity.EnderChestBlockEntity;
 import net.minecraft.client.render.entity.model.ChestBlockEntityModel;
 import net.minecraft.util.Identifier;
 
 public class EnderChestBlockEntityRenderer extends BlockEntityRenderer<EnderChestBlockEntity> {
