--- TeleportToSpecificPlayerSpectatorCommand.java
+++ TeleportToSpecificPlayerSpectatorCommand.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.gui.hud.spectator;
 
 import com.mojang.authlib.GameProfile;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.DrawableHelper;
 import net.minecraft.client.network.AbstractClientPlayerEntity;
 import net.minecraft.network.packet.c2s.play.SpectatorTeleportC2SPacket;
 import net.minecraft.text.LiteralText;
