--- ControlsOptionsScreen.java
+++ ControlsOptionsScreen.java
@@ -1,7 +1,8 @@
 package net.minecraft.client.gui.screen.options;
 
+import com.element.render.MatrixStack;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.gui.widget.OptionButtonWidget;
 import net.minecraft.client.gui.widget.OptionSliderWidget;
@@ -104,11 +105,11 @@
 			super.keyPressed(c, i);
 		}
 	}
 
 	@Override
-	public void render(int i, int j, float f) {
+	public void render(MatrixStack matrices, int j, float f, int i) {
 		this.renderBackground();
 		this.keyBindingListWidget.render(i, j, f);
 		this.drawCenteredString(this.textRenderer, this.title, this.width / 2, 8, 16777215);
 		boolean bl = true;
 
@@ -118,8 +119,8 @@
 				break;
 			}
 		}
 
 		this.resetButton.active = !bl;
-		super.render(i, j, f);
+		super.render(matrices, j, f, i);
 	}
 }
