--- AbstractMinecartEntity.java
+++ AbstractMinecartEntity.java
@@ -83,15 +83,15 @@
 		return false;
 	}
 
 	@Override
 	protected void initDataTracker() {
-		this.dataTracker.track(17, new Integer(0));
-		this.dataTracker.track(18, new Integer(1));
-		this.dataTracker.track(19, new Float(0.0F));
-		this.dataTracker.track(20, new Integer(0));
-		this.dataTracker.track(21, new Integer(6));
+		this.dataTracker.track(17, 0);
+		this.dataTracker.track(18, 1);
+		this.dataTracker.track(19, 0.0F);
+		this.dataTracker.track(20, 0);
+		this.dataTracker.track(21, 6);
 		this.dataTracker.track(22, (byte)0);
 	}
 
 	@Override
 	public Box getHardCollisionBox(Entity entity) {
@@ -280,11 +280,11 @@
 				if (this.yawFlipped) {
 					this.yaw += 180.0F;
 				}
 			}
 
-			double m = (double)MathHelper.wrapDegrees(this.yaw - this.prevYaw);
+			double m = (double) MathHelper.wrapDegrees(this.yaw - this.prevYaw);
 			if (m < -170.0 || m >= 170.0) {
 				this.yaw += 180.0F;
 				this.yawFlipped = !this.yawFlipped;
 			}
 
@@ -661,11 +661,11 @@
 
 					double d = entity.x - this.x;
 					double e = entity.z - this.z;
 					double f = d * d + e * e;
 					if (f >= 1.0E-4F) {
-						f = (double)MathHelper.sqrt(f);
+						f = (double) MathHelper.sqrt(f);
 						d /= f;
 						e /= f;
 						double g = 1.0 / f;
 						if (g > 1.0) {
 							g = 1.0;
@@ -681,11 +681,11 @@
 						e *= 0.5;
 						if (entity instanceof AbstractMinecartEntity) {
 							double h = entity.x - this.x;
 							double i = entity.z - this.z;
 							Vec3d vec3d = new Vec3d(h, 0.0, i).normalize();
-							Vec3d vec3d2 = new Vec3d((double)MathHelper.cos(this.yaw * (float) Math.PI / 180.0F), 0.0, (double)MathHelper.sin(this.yaw * (float) Math.PI / 180.0F))
+							Vec3d vec3d2 = new Vec3d((double) MathHelper.cos(this.yaw * (float) Math.PI / 180.0F), 0.0, (double) MathHelper.sin(this.yaw * (float) Math.PI / 180.0F))
 								.normalize();
 							double j = Math.abs(vec3d.dotProduct(vec3d2));
 							if (j < 0.8F) {
 								return;
 							}
