--- BlockPos.java
+++ BlockPos.java
@@ -1,10 +1,11 @@
 package net.minecraft.util.math;
 
 import com.google.common.collect.AbstractIterator;
 import java.util.Iterator;
 import net.minecraft.entity.Entity;
+import net.minecraft.util.math.MathHelper;
 
 public class BlockPos extends Vec3i {
 	public static final BlockPos ORIGIN = new BlockPos(0, 0, 0);
 	private static final int SIZE_BITS_X = 1 + MathHelper.log2(MathHelper.smallestEncompassingPowerOfTwo(30000000));
 	private static final int SIZE_BITS_Z = SIZE_BITS_X;
