--- Vec3d.java
+++ Vec3d.java
@@ -1,7 +1,9 @@
 package net.minecraft.util.math;
 
+import net.minecraft.util.math.MathHelper;
+
 public class Vec3d {
 	public final double x;
 	public final double y;
 	public final double z;
 
@@ -30,11 +32,11 @@
 	public Vec3d reverseSubtract(Vec3d vec3d) {
 		return new Vec3d(vec3d.x - this.x, vec3d.y - this.y, vec3d.z - this.z);
 	}
 
 	public Vec3d normalize() {
-		double d = (double)MathHelper.sqrt(this.x * this.x + this.y * this.y + this.z * this.z);
+		double d = (double) MathHelper.sqrt(this.x * this.x + this.y * this.y + this.z * this.z);
 		return d < 1.0E-4 ? new Vec3d(0.0, 0.0, 0.0) : new Vec3d(this.x / d, this.y / d, this.z / d);
 	}
 
 	public double dotProduct(Vec3d vec3d) {
 		return this.x * vec3d.x + this.y * vec3d.y + this.z * vec3d.z;
@@ -62,22 +64,22 @@
 
 	public double distanceTo(Vec3d vec3d) {
 		double d = vec3d.x - this.x;
 		double e = vec3d.y - this.y;
 		double f = vec3d.z - this.z;
-		return (double)MathHelper.sqrt(d * d + e * e + f * f);
+		return (double) MathHelper.sqrt(d * d + e * e + f * f);
 	}
 
 	public double squaredDistanceTo(Vec3d vec3d) {
 		double d = vec3d.x - this.x;
 		double e = vec3d.y - this.y;
 		double f = vec3d.z - this.z;
 		return d * d + e * e + f * f;
 	}
 
 	public double length() {
-		return (double)MathHelper.sqrt(this.x * this.x + this.y * this.y + this.z * this.z);
+		return (double) MathHelper.sqrt(this.x * this.x + this.y * this.y + this.z * this.z);
 	}
 
 	public Vec3d lerpForX(Vec3d vec3d, double d) {
 		double e = vec3d.x - this.x;
 		double f = vec3d.y - this.y;
