--- ListedChunkRenderManager.java
+++ ListedChunkRenderManager.java
@@ -1,8 +1,8 @@
 package net.minecraft.client.render.world;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GlStateManager;
 import net.minecraft.client.render.RenderLayer;
 import net.minecraft.client.world.BuiltChunk;
 import org.lwjgl.opengl.GL11;
 
 public class ListedChunkRenderManager extends AbstractChunkRenderManager {
