--- VboChunkRenderManager.java
+++ VboChunkRenderManager.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.render.world;
 
-import com.mojang.blaze3d.platform.GLX;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GLX;
+import com.mojang.blaze3d.platform.GlStateManager;
 import net.minecraft.client.render.RenderLayer;
 import net.minecraft.client.render.VertexBuffer;
 import net.minecraft.client.world.BuiltChunk;
 import org.lwjgl.opengl.GL11;
 
