--- TheEndDimension.java
+++ TheEndDimension.java
@@ -1,9 +1,9 @@
 package net.minecraft.world.dimension;
 
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.biome.SingletonBiomeSource;
 import net.minecraft.world.chunk.ChunkProvider;
 import net.minecraft.world.chunk.EndChunkGenerator;
@@ -32,12 +32,12 @@
 	}
 
 	@Override
 	public Vec3d getFogColor(float f, float g) {
 		int i = 10518688;
-		float h = MathHelper.cos(f * (float) Math.PI * 2.0F) * 2.0F + 0.5F;
-		h = MathHelper.clamp(h, 0.0F, 1.0F);
+		float h = Mth.cos(f * (float) Math.PI * 2.0F) * 2.0F + 0.5F;
+		h = Mth.clamp(h, 0.0F, 1.0F);
 		float j = (float)(i >> 16 & 0xFF) / 255.0F;
 		float k = (float)(i >> 8 & 0xFF) / 255.0F;
 		float l = (float)(i & 0xFF) / 255.0F;
 		j *= h * 0.0F + 0.15F;
 		k *= h * 0.0F + 0.15F;
