--- SpectatorHud.java
+++ SpectatorHud.java
@@ -1,19 +1,19 @@
 package net.minecraft.client.gui.hud;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.DrawableHelper;
 import net.minecraft.client.gui.hud.spectator.SpectatorMenu;
 import net.minecraft.client.gui.hud.spectator.SpectatorMenuCloseCallback;
 import net.minecraft.client.gui.hud.spectator.SpectatorMenuCommand;
 import net.minecraft.client.gui.hud.spectator.SpectatorMenuState;
 import net.minecraft.client.option.GameOptions;
 import net.minecraft.client.render.DiffuseLighting;
 import net.minecraft.client.util.Window;
 import net.minecraft.util.Identifier;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class SpectatorHud extends DrawableHelper implements SpectatorMenuCloseCallback {
 	private static final Identifier WIDGETS_TEXTURE = new Identifier("textures/gui/widgets.png");
 	public static final Identifier SPECTATOR_TEXTURE = new Identifier("textures/gui/spectator_widgets.png");
 	private final MinecraftClient client;
@@ -33,11 +33,11 @@
 		}
 	}
 
 	private float getSpectatorMenuHeight() {
 		long l = this.lastInteractionTime - MinecraftClient.getTime() + 5000L;
-		return MathHelper.clamp((float)l / 2000.0F, 0.0F, 1.0F);
+		return Mth.clamp((float)l / 2000.0F, 0.0F, 1.0F);
 	}
 
 	public void render(Window window, float f) {
 		if (this.spectatorMenu != null) {
 			float g = this.getSpectatorMenuHeight();
