--- MobEntity.java
+++ MobEntity.java
@@ -477,11 +477,11 @@
 			h = livingEntity.y + (double)livingEntity.getEyeHeight() - (this.y + (double)this.getEyeHeight());
 		} else {
 			h = (entity.getBoundingBox().minY + entity.getBoundingBox().maxY) / 2.0 - (this.y + (double)this.getEyeHeight());
 		}
 
-		double i = (double)MathHelper.sqrt(d * d + e * e);
+		double i = (double) MathHelper.sqrt(d * d + e * e);
 		float j = (float)(MathHelper.atan2(e, d) * 180.0 / (float) Math.PI) - 90.0F;
 		float k = (float)(-(MathHelper.atan2(h, i) * 180.0 / (float) Math.PI));
 		this.pitch = this.changeAngle(this.pitch, k, g);
 		this.yaw = this.changeAngle(this.yaw, j, f);
 	}
@@ -733,11 +733,11 @@
 	public boolean isPersistent() {
 		return this.persistent;
 	}
 
 	@Override
-	public final boolean openInventory(PlayerEntity playerEntity) {
+	public boolean openInventory(PlayerEntity playerEntity) {
 		if (this.isLeashed() && this.getLeashOwner() == playerEntity) {
 			this.detachLeash(true, !playerEntity.abilities.creativeMode);
 			return true;
 		} else {
 			ItemStack itemStack = playerEntity.inventory.getMainHandStack();
