--- EndCrystalEntity.java
+++ EndCrystalEntity.java
@@ -2,11 +2,11 @@
 
 import net.minecraft.block.Blocks;
 import net.minecraft.entity.damage.DamageSource;
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.dimension.TheEndDimension;
 
 public class EndCrystalEntity extends Entity {
 	public int endCrystalAge;
@@ -40,13 +40,13 @@
 		this.prevX = this.x;
 		this.prevY = this.y;
 		this.prevZ = this.z;
 		this.endCrystalAge++;
 		this.dataTracker.setProperty(8, this.field_3738);
-		int i = MathHelper.floor(this.x);
-		int j = MathHelper.floor(this.y);
-		int k = MathHelper.floor(this.z);
+		int i = Mth.floor(this.x);
+		int j = Mth.floor(this.y);
+		int k = Mth.floor(this.z);
 		if (this.world.dimension instanceof TheEndDimension && this.world.getBlockState(new BlockPos(i, j, k)).getBlock() != Blocks.FIRE) {
 			this.world.setBlockState(new BlockPos(i, j, k), Blocks.FIRE.getDefaultState());
 		}
 	}
 
