--- ExperienceOrbEntity.java
+++ ExperienceOrbEntity.java
@@ -3,11 +3,11 @@
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.damage.DamageSource;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class ExperienceOrbEntity extends Entity {
 	public int renderTicks;
 	public int orbAge;
@@ -43,11 +43,11 @@
 	}
 
 	@Override
 	public int getLightmapCoordinates(float f) {
 		float g = 0.5F;
-		g = MathHelper.clamp(g, 0.0F, 1.0F);
+		g = Mth.clamp(g, 0.0F, 1.0F);
 		int i = super.getLightmapCoordinates(f);
 		int j = i & 0xFF;
 		int k = i >> 16 & 0xFF;
 		j += (int)(g * 15.0F * 16.0F);
 		if (j > 240) {
@@ -104,11 +104,11 @@
 		}
 
 		this.move(this.velocityX, this.velocityY, this.velocityZ);
 		float j = 0.98F;
 		if (this.onGround) {
-			j = this.world.getBlockState(new BlockPos(MathHelper.floor(this.x), MathHelper.floor(this.getBoundingBox().minY) - 1, MathHelper.floor(this.z))).getBlock().slipperiness
+			j = this.world.getBlockState(new BlockPos(Mth.floor(this.x), Mth.floor(this.getBoundingBox().minY) - 1, Mth.floor(this.z))).getBlock().slipperiness
 				* 0.98F;
 		}
 
 		this.velocityX *= (double)j;
 		this.velocityY *= 0.98F;
