--- FallingBlockEntity.java
+++ FallingBlockEntity.java
@@ -16,11 +16,11 @@
 import net.minecraft.nbt.NbtElement;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.crash.CrashReportSection;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class FallingBlockEntity extends Entity {
 	private BlockState block;
 	public int timeFalling;
@@ -138,18 +138,18 @@
 
 	@Override
 	public void handleFallDamage(float f, float g) {
 		Block block = this.block.getBlock();
 		if (this.hurtEntities) {
-			int i = MathHelper.ceil(f - 1.0F);
+			int i = Mth.ceil(f - 1.0F);
 			if (i > 0) {
 				List<Entity> list = Lists.newArrayList(this.world.getEntitiesIn(this, this.getBoundingBox()));
 				boolean bl = block == Blocks.ANVIL;
 				DamageSource damageSource = bl ? DamageSource.ANVIL : DamageSource.FALLING_BLOCK;
 
 				for (Entity entity : list) {
-					entity.damage(damageSource, (float)Math.min(MathHelper.floor((float)i * this.fallHurtAmount), this.fallHurtMax));
+					entity.damage(damageSource, (float)Math.min(Mth.floor((float)i * this.fallHurtAmount), this.fallHurtMax));
 				}
 
 				if (bl && (double)this.random.nextFloat() < 0.05F + (double)i * 0.05) {
 					int j = (Integer)this.block.get(AnvilBlock.DAMAGE);
 					if (++j > 2) {
