--- FireworkRocketEntity.java
+++ FireworkRocketEntity.java
@@ -1,11 +1,11 @@
 package net.minecraft.entity;
 
 import net.minecraft.client.particle.ParticleType;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NbtCompound;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class FireworkRocketEntity extends Entity {
 	private int life;
 	private int lifeTime;
@@ -50,13 +50,13 @@
 	public void setVelocityClient(double d, double e, double f) {
 		this.velocityX = d;
 		this.velocityY = e;
 		this.velocityZ = f;
 		if (this.prevPitch == 0.0F && this.prevYaw == 0.0F) {
-			float g = MathHelper.sqrt(d * d + f * f);
-			this.prevYaw = this.yaw = (float)(MathHelper.atan2(d, f) * 180.0 / (float) Math.PI);
-			this.prevPitch = this.pitch = (float)(MathHelper.atan2(e, (double)g) * 180.0 / (float) Math.PI);
+			float g = Mth.sqrt(d * d + f * f);
+			this.prevYaw = this.yaw = (float)(Mth.atan2(d, f) * 180.0 / (float) Math.PI);
+			this.prevPitch = this.pitch = (float)(Mth.atan2(e, (double)g) * 180.0 / (float) Math.PI);
 		}
 	}
 
 	@Override
 	public void tick() {
@@ -66,13 +66,13 @@
 		super.tick();
 		this.velocityX *= 1.15;
 		this.velocityZ *= 1.15;
 		this.velocityY += 0.04;
 		this.move(this.velocityX, this.velocityY, this.velocityZ);
-		float f = MathHelper.sqrt(this.velocityX * this.velocityX + this.velocityZ * this.velocityZ);
-		this.yaw = (float)(MathHelper.atan2(this.velocityX, this.velocityZ) * 180.0 / (float) Math.PI);
-		this.pitch = (float)(MathHelper.atan2(this.velocityY, (double)f) * 180.0 / (float) Math.PI);
+		float f = Mth.sqrt(this.velocityX * this.velocityX + this.velocityZ * this.velocityZ);
+		this.yaw = (float)(Mth.atan2(this.velocityX, this.velocityZ) * 180.0 / (float) Math.PI);
+		this.pitch = (float)(Mth.atan2(this.velocityY, (double)f) * 180.0 / (float) Math.PI);
 
 		while (this.pitch - this.prevPitch < -180.0F) {
 			this.prevPitch -= 360.0F;
 		}
 
