--- LivingEntity.java
+++ LivingEntity.java
@@ -149,11 +149,11 @@
 		}
 
 		if (!this.world.isClient && this.fallDistance > 3.0F && bl) {
 			BlockState blockState = this.world.getBlockState(blockPos);
 			Block block2 = blockState.getBlock();
-			float f = (float)MathHelper.ceil(this.fallDistance - 3.0F);
+			float f = (float) MathHelper.ceil(this.fallDistance - 3.0F);
 			if (block2.getMaterial() != Material.AIR) {
 				double e = (double)Math.min(0.2F + f / 15.0F, 10.0F);
 				if (e > 2.5) {
 					e = 2.5;
 				}
@@ -1286,11 +1286,11 @@
 		this.prevStepBobbingAmount = this.stepBobbingAmount;
 		float k = 0.0F;
 		if (f > 0.0025000002F) {
 			k = 1.0F;
 			h = (float)Math.sqrt((double)f) * 3.0F;
-			g = (float)MathHelper.atan2(e, d) * 180.0F / (float) Math.PI - 90.0F;
+			g = (float) MathHelper.atan2(e, d) * 180.0F / (float) Math.PI - 90.0F;
 		}
 
 		if (this.handSwingProgress > 0.0F) {
 			g = this.yaw;
 		}
