--- PortalTeleporter.java
+++ PortalTeleporter.java
@@ -11,11 +11,11 @@
 import net.minecraft.server.world.ServerWorld;
 import net.minecraft.util.collection.LongObjectStorage;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.ChunkPos;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class PortalTeleporter {
 	private final ServerWorld world;
 	private final Random random;
 	private final LongObjectStorage<PortalTeleporter.Position> field_5472 = new LongObjectStorage<>();
@@ -31,13 +31,13 @@
 			if (!this.method_8584(entity, f)) {
 				this.method_3803(entity);
 				this.method_8584(entity, f);
 			}
 		} else {
-			int i = MathHelper.floor(entity.x);
-			int j = MathHelper.floor(entity.y) - 1;
-			int k = MathHelper.floor(entity.z);
+			int i = Mth.floor(entity.x);
+			int j = Mth.floor(entity.y) - 1;
+			int k = Mth.floor(entity.z);
 			int l = 1;
 			int m = 0;
 
 			for (int n = -2; n <= 2; n++) {
 				for (int o = -2; o <= 2; o++) {
@@ -57,12 +57,12 @@
 	}
 
 	public boolean method_8584(Entity entity, float f) {
 		int i = 128;
 		double d = -1.0;
-		int j = MathHelper.floor(entity.x);
-		int k = MathHelper.floor(entity.z);
+		int j = Mth.floor(entity.x);
+		int k = Mth.floor(entity.z);
 		boolean bl = true;
 		BlockPos blockPos = BlockPos.ORIGIN;
 		long l = ChunkPos.getIdFromCoords(j, k);
 		if (this.field_5472.contains(l)) {
 			PortalTeleporter.Position position = this.field_5472.get(l);
@@ -157,13 +157,13 @@
 	}
 
 	public boolean method_3803(Entity entity) {
 		int i = 16;
 		double d = -1.0;
-		int j = MathHelper.floor(entity.x);
-		int k = MathHelper.floor(entity.y);
-		int l = MathHelper.floor(entity.z);
+		int j = Mth.floor(entity.x);
+		int k = Mth.floor(entity.y);
+		int l = Mth.floor(entity.z);
 		int m = j;
 		int n = k;
 		int o = l;
 		int p = 0;
 		int q = this.random.nextInt(4);
@@ -274,11 +274,11 @@
 			af = -af;
 			ag = -ag;
 		}
 
 		if (d < 0.0) {
-			n = MathHelper.clamp(n, 70, this.world.getEffectiveHeight() - 10);
+			n = Mth.clamp(n, 70, this.world.getEffectiveHeight() - 10);
 			ae = n;
 
 			for (int txx = -1; txx <= 1; txx++) {
 				for (int u = 1; u < 3; u++) {
 					for (int vx = -1; vx < 3; vx++) {
