--- TrackedEntityInstance.java
+++ TrackedEntityInstance.java
@@ -50,11 +50,11 @@
 import net.minecraft.network.packet.s2c.play.MobSpawnS2CPacket;
 import net.minecraft.network.packet.s2c.play.PaintingSpawnS2CPacket;
 import net.minecraft.network.packet.s2c.play.PlayerSpawnS2CPacket;
 import net.minecraft.network.packet.s2c.play.UpdateEntityNbtS2CPacket;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class TrackedEntityInstance {
 	private static final Logger LOGGER = LogManager.getLogger();
@@ -86,16 +86,16 @@
 	public TrackedEntityInstance(Entity entity, int i, int j, boolean bl) {
 		this.trackedEntity = entity;
 		this.trackingDistance = i;
 		this.tracingFrequency = j;
 		this.trackVelocity = bl;
-		this.serializedX = MathHelper.floor(entity.x * 32.0);
-		this.serializedY = MathHelper.floor(entity.y * 32.0);
-		this.serializedZ = MathHelper.floor(entity.z * 32.0);
-		this.serializedYaw = MathHelper.floor(entity.yaw * 256.0F / 360.0F);
-		this.serializedPitch = MathHelper.floor(entity.pitch * 256.0F / 360.0F);
-		this.headRotationYaw = MathHelper.floor(entity.getHeadRotation() * 256.0F / 360.0F);
+		this.serializedX = Mth.floor(entity.x * 32.0);
+		this.serializedY = Mth.floor(entity.y * 32.0);
+		this.serializedZ = Mth.floor(entity.z * 32.0);
+		this.serializedYaw = Mth.floor(entity.yaw * 256.0F / 360.0F);
+		this.serializedPitch = Mth.floor(entity.pitch * 256.0F / 360.0F);
+		this.headRotationYaw = Mth.floor(entity.getHeadRotation() * 256.0F / 360.0F);
 		this.onGround = entity.onGround;
 	}
 
 	public boolean equals(Object object) {
 		return object instanceof TrackedEntityInstance ? ((TrackedEntityInstance)object).trackedEntity.getEntityId() == this.trackedEntity.getEntityId() : false;
@@ -141,15 +141,15 @@
 		}
 
 		if (this.field_2871 % this.tracingFrequency == 0 || this.trackedEntity.velocityDirty || this.trackedEntity.getDataTracker().isDirty()) {
 			if (this.trackedEntity.vehicle == null) {
 				this.field_2879++;
-				int i = MathHelper.floor(this.trackedEntity.x * 32.0);
-				int j = MathHelper.floor(this.trackedEntity.y * 32.0);
-				int k = MathHelper.floor(this.trackedEntity.z * 32.0);
-				int l = MathHelper.floor(this.trackedEntity.yaw * 256.0F / 360.0F);
-				int m = MathHelper.floor(this.trackedEntity.pitch * 256.0F / 360.0F);
+				int i = Mth.floor(this.trackedEntity.x * 32.0);
+				int j = Mth.floor(this.trackedEntity.y * 32.0);
+				int k = Mth.floor(this.trackedEntity.z * 32.0);
+				int l = Mth.floor(this.trackedEntity.yaw * 256.0F / 360.0F);
+				int m = Mth.floor(this.trackedEntity.pitch * 256.0F / 360.0F);
 				int n = i - this.serializedX;
 				int o = j - this.serializedY;
 				int p = k - this.serializedZ;
 				Packet packet2 = null;
 				boolean bl = Math.abs(n) >= 4 || Math.abs(o) >= 4 || Math.abs(p) >= 4 || this.field_2871 % 60 == 0;
@@ -212,27 +212,27 @@
 					this.serializedPitch = m;
 				}
 
 				this.field_5303 = false;
 			} else {
-				int ix = MathHelper.floor(this.trackedEntity.yaw * 256.0F / 360.0F);
-				int jx = MathHelper.floor(this.trackedEntity.pitch * 256.0F / 360.0F);
+				int ix = Mth.floor(this.trackedEntity.yaw * 256.0F / 360.0F);
+				int jx = Mth.floor(this.trackedEntity.pitch * 256.0F / 360.0F);
 				boolean bl3 = Math.abs(ix - this.serializedYaw) >= 4 || Math.abs(jx - this.serializedPitch) >= 4;
 				if (bl3) {
 					this.method_2179(new EntityS2CPacket.Rotate(this.trackedEntity.getEntityId(), (byte)ix, (byte)jx, this.trackedEntity.onGround));
 					this.serializedYaw = ix;
 					this.serializedPitch = jx;
 				}
 
-				this.serializedX = MathHelper.floor(this.trackedEntity.x * 32.0);
-				this.serializedY = MathHelper.floor(this.trackedEntity.y * 32.0);
-				this.serializedZ = MathHelper.floor(this.trackedEntity.z * 32.0);
+				this.serializedX = Mth.floor(this.trackedEntity.x * 32.0);
+				this.serializedY = Mth.floor(this.trackedEntity.y * 32.0);
+				this.serializedZ = Mth.floor(this.trackedEntity.z * 32.0);
 				this.method_6068();
 				this.field_5303 = true;
 			}
 
-			int ix = MathHelper.floor(this.trackedEntity.getHeadRotation() * 256.0F / 360.0F);
+			int ix = Mth.floor(this.trackedEntity.getHeadRotation() * 256.0F / 360.0F);
 			if (Math.abs(ix - this.headRotationYaw) >= 4) {
 				this.method_2179(new EntitySetHeadYawS2CPacket(this.trackedEntity, (byte)ix));
 				this.headRotationYaw = ix;
 			}
 
@@ -398,11 +398,11 @@
 			AbstractMinecartEntity abstractMinecartEntity = (AbstractMinecartEntity)this.trackedEntity;
 			return new EntitySpawnS2CPacket(this.trackedEntity, 10, abstractMinecartEntity.getMinecartType().getId());
 		} else if (this.trackedEntity instanceof BoatEntity) {
 			return new EntitySpawnS2CPacket(this.trackedEntity, 1);
 		} else if (this.trackedEntity instanceof EntityCategoryProvider) {
-			this.headRotationYaw = MathHelper.floor(this.trackedEntity.getHeadRotation() * 256.0F / 360.0F);
+			this.headRotationYaw = Mth.floor(this.trackedEntity.getHeadRotation() * 256.0F / 360.0F);
 			return new MobSpawnS2CPacket((LivingEntity)this.trackedEntity);
 		} else if (this.trackedEntity instanceof FishingBobberEntity) {
 			Entity entity = ((FishingBobberEntity)this.trackedEntity).thrower;
 			return new EntitySpawnS2CPacket(this.trackedEntity, 90, entity != null ? entity.getEntityId() : this.trackedEntity.getEntityId());
 		} else if (this.trackedEntity instanceof AbstractArrowEntity) {
@@ -455,21 +455,21 @@
 			return new PaintingSpawnS2CPacket((PaintingEntity)this.trackedEntity);
 		} else if (this.trackedEntity instanceof ItemFrameEntity) {
 			ItemFrameEntity itemFrameEntity = (ItemFrameEntity)this.trackedEntity;
 			EntitySpawnS2CPacket entitySpawnS2CPacketx = new EntitySpawnS2CPacket(this.trackedEntity, 71, itemFrameEntity.direction.getHorizontal());
 			BlockPos blockPos = itemFrameEntity.getTilePos();
-			entitySpawnS2CPacketx.setX(MathHelper.floor((float)(blockPos.getX() * 32)));
-			entitySpawnS2CPacketx.setY(MathHelper.floor((float)(blockPos.getY() * 32)));
-			entitySpawnS2CPacketx.setZ(MathHelper.floor((float)(blockPos.getZ() * 32)));
+			entitySpawnS2CPacketx.setX(Mth.floor((float)(blockPos.getX() * 32)));
+			entitySpawnS2CPacketx.setY(Mth.floor((float)(blockPos.getY() * 32)));
+			entitySpawnS2CPacketx.setZ(Mth.floor((float)(blockPos.getZ() * 32)));
 			return entitySpawnS2CPacketx;
 		} else if (this.trackedEntity instanceof LeashKnotEntity) {
 			LeashKnotEntity leashKnotEntity = (LeashKnotEntity)this.trackedEntity;
 			EntitySpawnS2CPacket entitySpawnS2CPacketx = new EntitySpawnS2CPacket(this.trackedEntity, 77);
 			BlockPos blockPos = leashKnotEntity.getTilePos();
-			entitySpawnS2CPacketx.setX(MathHelper.floor((float)(blockPos.getX() * 32)));
-			entitySpawnS2CPacketx.setY(MathHelper.floor((float)(blockPos.getY() * 32)));
-			entitySpawnS2CPacketx.setZ(MathHelper.floor((float)(blockPos.getZ() * 32)));
+			entitySpawnS2CPacketx.setX(Mth.floor((float)(blockPos.getX() * 32)));
+			entitySpawnS2CPacketx.setY(Mth.floor((float)(blockPos.getY() * 32)));
+			entitySpawnS2CPacketx.setZ(Mth.floor((float)(blockPos.getZ() * 32)));
 			return entitySpawnS2CPacketx;
 		} else if (this.trackedEntity instanceof ExperienceOrbEntity) {
 			return new ExperienceOrbSpawnS2CPacket((ExperienceOrbEntity)this.trackedEntity);
 		} else {
 			throw new IllegalArgumentException("Don't know how to add " + this.trackedEntity.getClass() + "!");
