--- AttributeModifier.java
+++ AttributeModifier.java
@@ -1,21 +1,21 @@
 package net.minecraft.entity.attribute;
 
 import io.netty.util.internal.ThreadLocalRandom;
 import java.util.UUID;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import org.apache.commons.lang3.Validate;
 
 public class AttributeModifier {
 	private final double amount;
 	private final int operation;
 	private final String name;
 	private final UUID id;
 	private boolean serialized = true;
 
 	public AttributeModifier(String string, double d, int i) {
-		this(MathHelper.randomUuid(ThreadLocalRandom.current()), string, d, i);
+		this(Mth.randomUuid(ThreadLocalRandom.current()), string, d, i);
 	}
 
 	public AttributeModifier(UUID uUID, String string, double d, int i) {
 		this.id = uUID;
 		this.name = string;
