--- IronGolemEntity.java
+++ IronGolemEntity.java
@@ -30,11 +30,11 @@
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.Item;
 import net.minecraft.item.Items;
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.village.Village;
 import net.minecraft.world.World;
 
 public class IronGolemEntity extends GolemEntity {
 	private int field_3724;
@@ -112,13 +112,13 @@
 		if (this.lookingAtVillagerTicksLeft > 0) {
 			this.lookingAtVillagerTicksLeft--;
 		}
 
 		if (this.velocityX * this.velocityX + this.velocityZ * this.velocityZ > 2.5000003E-7F && this.random.nextInt(5) == 0) {
-			int i = MathHelper.floor(this.x);
-			int j = MathHelper.floor(this.y - 0.2F);
-			int k = MathHelper.floor(this.z);
+			int i = Mth.floor(this.x);
+			int j = Mth.floor(this.y - 0.2F);
+			int k = Mth.floor(this.z);
 			BlockState blockState = this.world.getBlockState(new BlockPos(i, j, k));
 			Block block = blockState.getBlock();
 			if (block.getMaterial() != Material.AIR) {
 				this.world
 					.addParticle(
