--- RabbitEntity.java
+++ RabbitEntity.java
@@ -29,11 +29,11 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.util.CommonI18n;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.LocalDifficulty;
 import net.minecraft.world.World;
 
 public class RabbitEntity extends AnimalEntity {
@@ -174,11 +174,11 @@
 	@Override
 	public void attemptSprintingParticles() {
 	}
 
 	private void lookTowards(double d, double e) {
-		this.yaw = (float)(MathHelper.atan2(e - this.z, d - this.x) * 180.0 / (float) Math.PI) - 90.0F;
+		this.yaw = (float)(Mth.atan2(e - this.z, d - this.x) * 180.0 / (float) Math.PI) - 90.0F;
 	}
 
 	private void method_11086() {
 		((RabbitEntity.RabbitJumpControl)this.jumpControl).method_11097(true);
 	}
