--- SnowGolemEntity.java
+++ SnowGolemEntity.java
@@ -17,11 +17,11 @@
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.entity.thrown.SnowballEntity;
 import net.minecraft.item.Item;
 import net.minecraft.item.Items;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class SnowGolemEntity extends GolemEntity implements RangedAttackMob {
 	public SnowGolemEntity(World world) {
 		super(world);
@@ -43,25 +43,25 @@
 
 	@Override
 	public void tickMovement() {
 		super.tickMovement();
 		if (!this.world.isClient) {
-			int i = MathHelper.floor(this.x);
-			int j = MathHelper.floor(this.y);
-			int k = MathHelper.floor(this.z);
+			int i = Mth.floor(this.x);
+			int j = Mth.floor(this.y);
+			int k = Mth.floor(this.z);
 			if (this.tickFire()) {
 				this.damage(DamageSource.DROWN, 1.0F);
 			}
 
 			if (this.world.getBiome(new BlockPos(i, 0, k)).getTemperature(new BlockPos(i, j, k)) > 1.0F) {
 				this.damage(DamageSource.ON_FIRE, 1.0F);
 			}
 
 			for (int l = 0; l < 4; l++) {
-				i = MathHelper.floor(this.x + (double)((float)(l % 2 * 2 - 1) * 0.25F));
-				j = MathHelper.floor(this.y);
-				k = MathHelper.floor(this.z + (double)((float)(l / 2 % 2 * 2 - 1) * 0.25F));
+				i = Mth.floor(this.x + (double)((float)(l % 2 * 2 - 1) * 0.25F));
+				j = Mth.floor(this.y);
+				k = Mth.floor(this.z + (double)((float)(l / 2 % 2 * 2 - 1) * 0.25F));
 				BlockPos blockPos = new BlockPos(i, j, k);
 				if (this.world.getBlockState(blockPos).getBlock().getMaterial() == Material.AIR
 					&& this.world.getBiome(new BlockPos(i, 0, k)).getTemperature(blockPos) < 0.8F
 					&& Blocks.SNOW_LAYER.canBePlacedAtPos(this.world, blockPos)) {
 					this.world.setBlockState(blockPos, Blocks.SNOW_LAYER.getDefaultState());
@@ -89,11 +89,11 @@
 		SnowballEntity snowballEntity = new SnowballEntity(this.world, this);
 		double d = livingEntity.y + (double)livingEntity.getEyeHeight() - 1.1F;
 		double e = livingEntity.x - this.x;
 		double g = d - snowballEntity.y;
 		double h = livingEntity.z - this.z;
-		float i = MathHelper.sqrt(e * e + h * h) * 0.2F;
+		float i = Mth.sqrt(e * e + h * h) * 0.2F;
 		snowballEntity.setVelocity(e, g + (double)i, h, 1.6F, 12.0F);
 		this.playSound("random.bow", 1.0F, 1.0F / (this.getRandom().nextFloat() * 0.4F + 0.8F));
 		this.world.spawnEntity(snowballEntity);
 	}
 
