--- SquidEntity.java
+++ SquidEntity.java
@@ -132,14 +132,14 @@
 				this.velocityY = (double)(this.constantVelocityY * this.constantVelocityRate);
 				this.velocityZ = (double)(this.constantVelocityZ * this.constantVelocityRate);
 			}
 
 			float f = MathHelper.sqrt(this.velocityX * this.velocityX + this.velocityZ * this.velocityZ);
-			this.bodyYaw = this.bodyYaw + (-((float)MathHelper.atan2(this.velocityX, this.velocityZ)) * 180.0F / (float) Math.PI - this.bodyYaw) * 0.1F;
+			this.bodyYaw = this.bodyYaw + (-((float) MathHelper.atan2(this.velocityX, this.velocityZ)) * 180.0F / (float) Math.PI - this.bodyYaw) * 0.1F;
 			this.yaw = this.bodyYaw;
 			this.rollAngle = (float)((double)this.rollAngle + Math.PI * (double)this.turningSpeed * 1.5);
-			this.tiltAngle = this.tiltAngle + (-((float)MathHelper.atan2((double)f, this.velocityY)) * 180.0F / (float) Math.PI - this.tiltAngle) * 0.1F;
+			this.tiltAngle = this.tiltAngle + (-((float) MathHelper.atan2((double)f, this.velocityY)) * 180.0F / (float) Math.PI - this.tiltAngle) * 0.1F;
 		} else {
 			this.tentacleAngle = MathHelper.abs(MathHelper.sin(this.thrustTimer)) * (float) Math.PI * 0.25F;
 			if (!this.world.isClient) {
 				this.velocityX = 0.0;
 				this.velocityY -= 0.08;
