--- VillagerEntity.java
+++ VillagerEntity.java
@@ -51,11 +51,11 @@
 import net.minecraft.text.Text;
 import net.minecraft.text.TranslatableText;
 import net.minecraft.util.DyeColor;
 import net.minecraft.util.Pair;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.village.TradeOffer;
 import net.minecraft.village.TraderOfferList;
 import net.minecraft.village.Village;
 import net.minecraft.world.LocalDifficulty;
 import net.minecraft.world.World;
@@ -901,11 +901,11 @@
 		}
 
 		@Override
 		public void convert(TraderOfferList traderOfferList, Random random) {
 			Enchantment enchantment = Enchantment.ALL_ENCHANTMENTS[random.nextInt(Enchantment.ALL_ENCHANTMENTS.length)];
-			int i = MathHelper.nextInt(random, enchantment.getMinimumLevel(), enchantment.getMaximumLevel());
+			int i = Mth.nextInt(random, enchantment.getMinimumLevel(), enchantment.getMaximumLevel());
 			ItemStack itemStack = Items.ENCHANTED_BOOK.getAsItemStack(new EnchantmentLevelEntry(enchantment, i));
 			int j = 2 + random.nextInt(5 + i * 10) + 3 * i;
 			if (j > 64) {
 				j = 64;
 			}
