--- SoundSystem.java
+++ SoundSystem.java
@@ -315,15 +315,15 @@
 			}
 		}
 	}
 
 	private float method_7099(SoundInstance soundInstance, Sound sound) {
-		return (float)MathHelper.clamp((double)soundInstance.getPitch() * sound.getVolume(), 0.5, 2.0);
+		return (float) MathHelper.clamp((double)soundInstance.getPitch() * sound.getVolume(), 0.5, 2.0);
 	}
 
 	private float method_7100(SoundInstance soundInstance, Sound sound, SoundCategory soundCategory) {
-		return (float)MathHelper.clamp((double)soundInstance.getVolume() * sound.getPitch(), 0.0, 1.0) * this.getSoundVolume(soundCategory);
+		return (float) MathHelper.clamp((double)soundInstance.getVolume() * sound.getPitch(), 0.0, 1.0) * this.getSoundVolume(soundCategory);
 	}
 
 	public void pauseAll() {
 		for (String string : this.field_8195.keySet()) {
 			LOGGER.debug(MARKER, "Pausing channel {}", new Object[]{string});
