--- EndChunkGenerator.java
+++ EndChunkGenerator.java
@@ -7,11 +7,11 @@
 import net.minecraft.block.FallingBlock;
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.EntityCategory;
 import net.minecraft.util.ProgressListener;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.gen.NoiseGenerator;
 
 public class EndChunkGenerator implements ChunkProvider {
@@ -169,11 +169,11 @@
 
 		for (int p = 0; p < l; p++) {
 			for (int q = 0; q < n; q++) {
 				float f = (float)(p + i) / 1.0F;
 				float g = (float)(q + k) / 1.0F;
-				float h = 100.0F - MathHelper.sqrt(f * f + g * g) * 8.0F;
+				float h = 100.0F - Mth.sqrt(f * f + g * g) * 8.0F;
 				if (h > 80.0F) {
 					h = 80.0F;
 				}
 
 				if (h < -100.0F) {
@@ -196,11 +196,11 @@
 					s -= 8.0;
 					s += (double)h;
 					int w = 2;
 					if (r > m / 2 - w) {
 						double x = (double)((float)(r - (m / 2 - w)) / 64.0F);
-						x = MathHelper.clamp(x, 0.0, 1.0);
+						x = Mth.clamp(x, 0.0, 1.0);
 						s = s * (1.0 - x) + -3000.0 * x;
 					}
 
 					int var34 = 8;
 					if (r < var34) {
