--- NetherChunkGenerator.java
+++ NetherChunkGenerator.java
@@ -10,11 +10,11 @@
 import net.minecraft.predicate.block.BlockPredicate;
 import net.minecraft.structure.NetherFortressStructure;
 import net.minecraft.util.ProgressListener;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.ChunkPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.gen.NoiseGenerator;
 import net.minecraft.world.gen.carver.Carver;
 import net.minecraft.world.gen.carver.NetherCaveCarver;
@@ -273,11 +273,11 @@
 						h = h * (1.0 - w) + -10.0 * w;
 					}
 
 					if ((double)r < g) {
 						double w = (g - (double)r) / 4.0;
-						w = MathHelper.clamp(w, 0.0, 1.0);
+						w = Mth.clamp(w, 0.0, 1.0);
 						h = h * (1.0 - w) + -10.0 * w;
 					}
 
 					ds[o] = h;
 					o++;
