--- PlayerEntity.java
+++ PlayerEntity.java
@@ -1344,11 +1344,11 @@
 			if (i > 0) {
 				if (this.vehicle instanceof AbstractMinecartEntity) {
 					this.incrementStat(Stats.CM_MINECART, i);
 					if (this.field_4011 == null) {
 						this.field_4011 = new BlockPos(this);
-					} else if (this.field_4011.squaredDistanceTo((double)MathHelper.floor(this.x), (double)MathHelper.floor(this.y), (double)MathHelper.floor(this.z))
+					} else if (this.field_4011.squaredDistanceTo((double) MathHelper.floor(this.x), (double) MathHelper.floor(this.y), (double) MathHelper.floor(this.z))
 						>= 1000000.0) {
 						this.incrementStat(AchievementsAndCriterions.ON_A_RAIL);
 					}
 				} else if (this.vehicle instanceof BoatEntity) {
 					this.incrementStat(Stats.CM_SAILED, i);
