--- EnchantingTableBlockEntity.java
+++ EnchantingTableBlockEntity.java
@@ -50,11 +50,11 @@
 		PlayerEntity playerEntity = this.world
 			.getClosestPlayer((double)((float)this.pos.getX() + 0.5F), (double)((float)this.pos.getY() + 0.5F), (double)((float)this.pos.getZ() + 0.5F), 3.0);
 		if (playerEntity != null) {
 			double d = playerEntity.x - (double)((float)this.pos.getX() + 0.5F);
 			double e = playerEntity.z - (double)((float)this.pos.getZ() + 0.5F);
-			this.closedBookAngle = (float)MathHelper.atan2(e, d);
+			this.closedBookAngle = (float) MathHelper.atan2(e, d);
 			this.nextPageTurningSpeed += 0.1F;
 			if (this.nextPageTurningSpeed < 0.5F || RANDOM.nextInt(40) == 0) {
 				float f = this.nextPageTurningVelocity;
 
 				do {
