--- HopperBlockEntity.java
+++ HopperBlockEntity.java
@@ -19,11 +19,11 @@
 import net.minecraft.util.HopperProvider;
 import net.minecraft.util.Tickable;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.Direction;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.World;
 
 public class HopperBlockEntity extends LockableContainerBlockEntity implements HopperProvider, Tickable {
 	private ItemStack[] items = new ItemStack[5];
 	private String customName;
@@ -449,13 +449,13 @@
 		return world.getEntitiesInBox(ItemEntity.class, new Box(d - 0.5, e - 0.5, f - 0.5, d + 0.5, e + 0.5, f + 0.5), EntityPredicate.VALID_ENTITY);
 	}
 
 	public static Inventory getInventoryAt(World world, double d, double e, double f) {
 		Inventory inventory = null;
-		int i = MathHelper.floor(d);
-		int j = MathHelper.floor(e);
-		int k = MathHelper.floor(f);
+		int i = Mth.floor(d);
+		int j = Mth.floor(e);
+		int k = Mth.floor(f);
 		BlockPos blockPos = new BlockPos(i, j, k);
 		Block block = world.getBlockState(blockPos).getBlock();
 		if (block.hasBlockEntity()) {
 			BlockEntity blockEntity = world.getBlockEntity(blockPos);
 			if (blockEntity instanceof Inventory) {
