--- NoteBlockBlockEntity.java
+++ NoteBlockBlockEntity.java
@@ -19,11 +19,11 @@
 
 	@Override
 	public void fromNbt(NbtCompound nbtCompound) {
 		super.fromNbt(nbtCompound);
 		this.note = nbtCompound.getByte("note");
-		this.note = (byte)MathHelper.clamp(this.note, 0, 24);
+		this.note = (byte) MathHelper.clamp(this.note, 0, 24);
 	}
 
 	public void increaseNote() {
 		this.note = (byte)((this.note + 1) % 25);
 		this.markDirty();
