--- RandomVectorGenerator.java
+++ RandomVectorGenerator.java
@@ -31,11 +31,11 @@
 		int m = 0;
 		float f = -99999.0F;
 		boolean bl2;
 		if (pathAwareEntity.hasPositionTarget()) {
 			double d = pathAwareEntity.getPositionTarget()
-					.squaredDistanceTo((double)MathHelper.floor(pathAwareEntity.x), (double)MathHelper.floor(pathAwareEntity.y), (double)MathHelper.floor(pathAwareEntity.z))
+					.squaredDistanceTo((double) MathHelper.floor(pathAwareEntity.x), (double) MathHelper.floor(pathAwareEntity.y), (double) MathHelper.floor(pathAwareEntity.z))
 				+ 4.0;
 			double e = (double)(pathAwareEntity.getPositionTargetRange() + (float)i);
 			bl2 = d < e * e;
 		} else {
 			bl2 = false;
