--- TwitchAuth.java
+++ TwitchAuth.java
@@ -3,12 +3,12 @@
 import com.google.common.base.Strings;
 import com.google.common.collect.Maps;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
 import com.mojang.authlib.properties.Property;
-import com.mojang.blaze3d.platform.GLX;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GLX;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import java.io.IOException;
 import java.net.URL;
 import java.net.URLEncoder;
 import java.util.Map;
 import java.util.Set;
