--- MinecraftServer.java
+++ MinecraftServer.java
@@ -51,11 +51,11 @@
 import net.minecraft.util.UserCache;
 import net.minecraft.util.Util;
 import net.minecraft.util.crash.CrashException;
 import net.minecraft.util.crash.CrashReport;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.util.profiler.Profiler;
 import net.minecraft.util.snooper.Snoopable;
 import net.minecraft.util.snooper.Snooper;
 import net.minecraft.world.DemoServerWorld;
@@ -496,11 +496,11 @@
 		this.tick();
 		if (l - this.lastPlayerSampleUpdate >= 5000000000L) {
 			this.lastPlayerSampleUpdate = l;
 			this.serverMetadata.setPlayers(new ServerMetadata.Players(this.getMaxPlayerCount(), this.getCurrentPlayerCount()));
 			GameProfile[] gameProfiles = new GameProfile[Math.min(this.getCurrentPlayerCount(), 12)];
-			int i = MathHelper.nextInt(this.random, 0, this.getCurrentPlayerCount() - gameProfiles.length);
+			int i = Mth.nextInt(this.random, 0, this.getCurrentPlayerCount() - gameProfiles.length);
 
 			for (int j = 0; j < gameProfiles.length; j++) {
 				gameProfiles[j] = ((ServerPlayerEntity)this.playerManager.getPlayers().get(i + j)).getGameProfile();
 			}
 
@@ -842,11 +842,11 @@
 		}
 
 		snooper.addGameInfo("uses_auth", this.onlineMode);
 		snooper.addGameInfo("gui_state", this.hasGui() ? "enabled" : "disabled");
 		snooper.addGameInfo("run_time", (getTimeMillis() - snooper.getStartTime()) / 60L * 1000L);
-		snooper.addGameInfo("avg_tick_ms", (int)(MathHelper.average(this.lastTickLengths) * 1.0E-6));
+		snooper.addGameInfo("avg_tick_ms", (int)(Mth.average(this.lastTickLengths) * 1.0E-6));
 		int i = 0;
 		if (this.worlds != null) {
 			for (int j = 0; j < this.worlds.length; j++) {
 				if (this.worlds[j] != null) {
 					ServerWorld serverWorld = this.worlds[j];
