--- PlayerManager.java
+++ PlayerManager.java
@@ -503,12 +503,12 @@
 		}
 
 		serverWorld.profiler.pop();
 		if (i != 1) {
 			serverWorld.profiler.push("placing");
-			d = (double)MathHelper.clamp((int)d, -29999872, 29999872);
-			e = (double)MathHelper.clamp((int)e, -29999872, 29999872);
+			d = (double) MathHelper.clamp((int)d, -29999872, 29999872);
+			e = (double) MathHelper.clamp((int)e, -29999872, 29999872);
 			if (entity.isAlive()) {
 				entity.refreshPositionAndAngles(d, entity.y, e, entity.yaw, entity.pitch);
 				serverWorld2.getPortalTeleporter().method_8583(entity, g);
 				serverWorld2.spawnEntity(entity);
 				serverWorld2.checkChunk(entity, false);
