--- PlayerControlGoal.java
+++ PlayerControlGoal.java
@@ -9,11 +9,11 @@
 import net.minecraft.entity.mob.MobEntity;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class PlayerControlGoal extends Goal {
 	private final MobEntity mob;
 	private final float speed;
 	private float field_5351;
@@ -45,56 +45,56 @@
 
 	@Override
 	public void tick() {
 		PlayerEntity playerEntity = (PlayerEntity)this.mob.rider;
 		PathAwareEntity pathAwareEntity = (PathAwareEntity)this.mob;
-		float f = MathHelper.wrapDegrees(playerEntity.yaw - this.mob.yaw) * 0.5F;
+		float f = Mth.wrapDegrees(playerEntity.yaw - this.mob.yaw) * 0.5F;
 		if (f > 5.0F) {
 			f = 5.0F;
 		}
 
 		if (f < -5.0F) {
 			f = -5.0F;
 		}
 
-		this.mob.yaw = MathHelper.wrapDegrees(this.mob.yaw + f);
+		this.mob.yaw = Mth.wrapDegrees(this.mob.yaw + f);
 		if (this.field_5351 < this.speed) {
 			this.field_5351 = this.field_5351 + (this.speed - this.field_5351) * 0.01F;
 		}
 
 		if (this.field_5351 > this.speed) {
 			this.field_5351 = this.speed;
 		}
 
-		int i = MathHelper.floor(this.mob.x);
-		int j = MathHelper.floor(this.mob.y);
-		int k = MathHelper.floor(this.mob.z);
+		int i = Mth.floor(this.mob.x);
+		int j = Mth.floor(this.mob.y);
+		int k = Mth.floor(this.mob.z);
 		float g = this.field_5351;
 		if (this.field_5352) {
 			if (this.field_5353++ > this.field_5354) {
 				this.field_5352 = false;
 			}
 
-			g += g * 1.15F * MathHelper.sin((float)this.field_5353 / (float)this.field_5354 * (float) Math.PI);
+			g += g * 1.15F * Mth.sin((float)this.field_5353 / (float)this.field_5354 * (float) Math.PI);
 		}
 
 		float h = 0.91F;
 		if (this.mob.onGround) {
-			h = this.mob.world.getBlockState(new BlockPos(MathHelper.floor((float)i), MathHelper.floor((float)j) - 1, MathHelper.floor((float)k))).getBlock().slipperiness
+			h = this.mob.world.getBlockState(new BlockPos(Mth.floor((float)i), Mth.floor((float)j) - 1, Mth.floor((float)k))).getBlock().slipperiness
 				* 0.91F;
 		}
 
 		float l = 0.16277136F / (h * h * h);
-		float m = MathHelper.sin(pathAwareEntity.yaw * (float) Math.PI / 180.0F);
-		float n = MathHelper.cos(pathAwareEntity.yaw * (float) Math.PI / 180.0F);
+		float m = Mth.sin(pathAwareEntity.yaw * (float) Math.PI / 180.0F);
+		float n = Mth.cos(pathAwareEntity.yaw * (float) Math.PI / 180.0F);
 		float o = pathAwareEntity.getMovementSpeed() * l;
 		float p = Math.max(g, 1.0F);
 		p = o / p;
 		float q = g * p;
 		float r = -(q * m);
 		float s = q * n;
-		if (MathHelper.abs(r) > MathHelper.abs(s)) {
+		if (Mth.abs(r) > Mth.abs(s)) {
 			if (r < 0.0F) {
 				r -= this.mob.width / 2.0F;
 			}
 
 			if (r > 0.0F) {
@@ -111,15 +111,15 @@
 			if (s > 0.0F) {
 				s += this.mob.width / 2.0F;
 			}
 		}
 
-		int t = MathHelper.floor(this.mob.x + (double)r);
-		int u = MathHelper.floor(this.mob.z + (double)s);
-		int v = MathHelper.floor(this.mob.width + 1.0F);
-		int w = MathHelper.floor(this.mob.height + playerEntity.height + 1.0F);
-		int x = MathHelper.floor(this.mob.width + 1.0F);
+		int t = Mth.floor(this.mob.x + (double)r);
+		int u = Mth.floor(this.mob.z + (double)s);
+		int v = Mth.floor(this.mob.width + 1.0F);
+		int w = Mth.floor(this.mob.height + playerEntity.height + 1.0F);
+		int x = Mth.floor(this.mob.width + 1.0F);
 		if (i != t || k != u) {
 			Block block = this.mob.world.getBlockState(new BlockPos(i, j, k)).getBlock();
 			boolean bl = !this.method_8381(block)
 				&& (block.getMaterial() != Material.AIR || !this.method_8381(this.mob.world.getBlockState(new BlockPos(i, j - 1, k)).getBlock()));
 			if (bl
