--- ListWidget.java
+++ ListWidget.java
@@ -1,14 +1,14 @@
 package net.minecraft.client.gui.widget;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.DrawableHelper;
 import net.minecraft.client.render.BufferBuilder;
 import net.minecraft.client.render.Tessellator;
 import net.minecraft.client.render.VertexFormats;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import org.lwjgl.input.Mouse;
 
 public abstract class ListWidget {
 	protected final MinecraftClient client;
 	protected int width;
@@ -104,11 +104,11 @@
 		this.homeButtonId = i;
 		this.endButtonId = j;
 	}
 
 	protected void capYPosition() {
-		this.scrollAmount = MathHelper.clamp(this.scrollAmount, 0.0F, (float)this.getMaxScroll());
+		this.scrollAmount = Mth.clamp(this.scrollAmount, 0.0F, (float)this.getMaxScroll());
 	}
 
 	public int getMaxScroll() {
 		return Math.max(0, this.getMaxPosition() - (this.yEnd - this.yStart - 4));
 	}
@@ -203,11 +203,11 @@
 			bufferBuilder.vertex((double)this.xStart, (double)(this.yEnd - o), 0.0).texture(0.0, 0.0).color(0, 0, 0, 0).next();
 			tessellator.draw();
 			int p = this.getMaxScroll();
 			if (p > 0) {
 				int q = (this.yEnd - this.yStart) * (this.yEnd - this.yStart) / this.getMaxPosition();
-				q = MathHelper.clamp(q, 32, this.yEnd - this.yStart - 8);
+				q = Mth.clamp(q, 32, this.yEnd - this.yStart - 8);
 				int r = (int)this.scrollAmount * (this.yEnd - this.yStart - q) / p + this.yStart;
 				if (r < this.yStart) {
 					r = this.yStart;
 				}
 
@@ -281,11 +281,11 @@
 						if (p < 1) {
 							p = 1;
 						}
 
 						int q = (int)((float)((this.yEnd - this.yStart) * (this.yEnd - this.yStart)) / (float)this.getMaxPosition());
-						q = MathHelper.clamp(q, 32, this.yEnd - this.yStart - 8);
+						q = Mth.clamp(q, 32, this.yEnd - this.yStart - 8);
 						this.scrollSpeed = this.scrollSpeed / ((float)(this.yEnd - this.yStart - q) / (float)p);
 					} else {
 						this.scrollSpeed = 1.0F;
 					}
 
