--- TextFieldWidget.java
+++ TextFieldWidget.java
@@ -1,18 +1,18 @@
 package net.minecraft.client.gui.widget;
 
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.font.TextRenderer;
 import net.minecraft.client.gui.DrawableHelper;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.render.BufferBuilder;
 import net.minecraft.client.render.Tessellator;
 import net.minecraft.client.render.VertexFormats;
 import net.minecraft.util.SharedConstants;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class TextFieldWidget extends DrawableHelper {
 	private final int id;
 	private final TextRenderer textRenderer;
 	public int x;
@@ -198,11 +198,11 @@
 	}
 
 	public void setCursor(int i) {
 		this.selectionStart = i;
 		int j = this.text.length();
-		this.selectionStart = MathHelper.clamp(this.selectionStart, 0, j);
+		this.selectionStart = Mth.clamp(this.selectionStart, 0, j);
 		this.setSelectionEnd(this.selectionStart);
 	}
 
 	public void setCursorToStart() {
 		this.setCursor(0);
@@ -504,11 +504,11 @@
 				this.firstCharacterIndex += i - l;
 			} else if (i <= this.firstCharacterIndex) {
 				this.firstCharacterIndex = this.firstCharacterIndex - (this.firstCharacterIndex - i);
 			}
 
-			this.firstCharacterIndex = MathHelper.clamp(this.firstCharacterIndex, 0, j);
+			this.firstCharacterIndex = Mth.clamp(this.firstCharacterIndex, 0, j);
 		}
 	}
 
 	public void setFocusUnlocked(boolean bl) {
 		this.focusUnlocked = bl;
