--- TextFieldWidget.java
+++ TextFieldWidget.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.gui.widget;
 
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.font.TextRenderer;
 import net.minecraft.client.gui.DrawableHelper;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.render.BufferBuilder;
 import net.minecraft.client.render.Tessellator;
