--- HelpCommand.java
+++ HelpCommand.java
@@ -16,11 +16,11 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.text.ClickEvent;
 import net.minecraft.text.TranslatableText;
 import net.minecraft.util.Formatting;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class HelpCommand extends AbstractCommand {
 	@Override
 	public String getCommandName() {
 		return "help";
@@ -55,11 +55,11 @@
 			Command command = (Command)map.get(strings[0]);
 			if (command != null) {
 				throw new IncorrectUsageException(command.getUsageTranslationKey(commandSource));
 			}
 
-			if (MathHelper.parseInt(strings[0], -1) != -1) {
+			if (Mth.parseInt(strings[0], -1) != -1) {
 				throw var12;
 			}
 
 			throw new NotFoundException();
 		}
