--- TeleportCommand.java
+++ TeleportCommand.java
@@ -106,12 +106,12 @@
 
 					entity.startRiding(null);
 					((ServerPlayerEntity)entity).networkHandler.teleportRequest(coordinate.getAmount(), coordinate2.getAmount(), coordinate3.getAmount(), f, g, set);
 					entity.setHeadYaw(f);
 				} else {
-					float h = (float)MathHelper.wrapDegrees(coordinate4.getResult());
-					float fx = (float)MathHelper.wrapDegrees(coordinate5.getResult());
+					float h = (float) MathHelper.wrapDegrees(coordinate4.getResult());
+					float fx = (float) MathHelper.wrapDegrees(coordinate5.getResult());
 					if (fx > 90.0F || fx < -90.0F) {
 						fx = MathHelper.wrapDegrees(180.0F - fx);
 						h = MathHelper.wrapDegrees(h + 180.0F);
 					}
 
