--- GameOptions.java
+++ GameOptions.java
@@ -22,11 +22,11 @@
 import net.minecraft.client.sound.SoundCategory;
 import net.minecraft.client.texture.SpriteAtlasTexture;
 import net.minecraft.client.twitch.TwitchAuth;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.network.packet.c2s.play.ClientSettingsC2SPacket;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 import net.minecraft.world.Difficulty;
 import org.apache.commons.lang3.ArrayUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.lwjgl.input.Keyboard;
@@ -1169,20 +1169,20 @@
 		public void setMaxValue(float f) {
 			this.max = f;
 		}
 
 		public float getRatio(float f) {
-			return MathHelper.clamp((this.adjust(f) - this.min) / (this.max - this.min), 0.0F, 1.0F);
+			return Mth.clamp((this.adjust(f) - this.min) / (this.max - this.min), 0.0F, 1.0F);
 		}
 
 		public float getValue(float f) {
-			return this.adjust(this.min + (this.max - this.min) * MathHelper.clamp(f, 0.0F, 1.0F));
+			return this.adjust(this.min + (this.max - this.min) * Mth.clamp(f, 0.0F, 1.0F));
 		}
 
 		public float adjust(float f) {
 			f = this.round(f);
-			return MathHelper.clamp(f, this.min, this.max);
+			return Mth.clamp(f, this.min, this.max);
 		}
 
 		protected float round(float f) {
 			if (this.step > 0.0F) {
 				f = this.step * (float)Math.round(f / this.step);
