--- ProtectionEnchantment.java
+++ ProtectionEnchantment.java
@@ -80,11 +80,11 @@
 	}
 
 	public static double method_4658(Entity entity, double d) {
 		int i = EnchantmentHelper.getLevel(Enchantment.BLAST_PROTECTION.id, entity.getArmorStacks());
 		if (i > 0) {
-			d -= (double)MathHelper.floor(d * (double)((float)i * 0.15F));
+			d -= (double) MathHelper.floor(d * (double)((float)i * 0.15F));
 		}
 
 		return d;
 	}
 }
