--- AbstractZombieModel.java
+++ AbstractZombieModel.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class AbstractZombieModel extends BiPedModel {
 	public AbstractZombieModel() {
 		this(0.0F, false);
 	}
@@ -17,21 +17,21 @@
 	}
 
 	@Override
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
 		super.setAngles(f, g, h, i, j, k, entity);
-		float l = MathHelper.sin(this.handSwingProgress * (float) Math.PI);
-		float m = MathHelper.sin((1.0F - (1.0F - this.handSwingProgress) * (1.0F - this.handSwingProgress)) * (float) Math.PI);
+		float l = Mth.sin(this.handSwingProgress * (float) Math.PI);
+		float m = Mth.sin((1.0F - (1.0F - this.handSwingProgress) * (1.0F - this.handSwingProgress)) * (float) Math.PI);
 		this.rightArm.posZ = 0.0F;
 		this.leftArm.posZ = 0.0F;
 		this.rightArm.posY = -(0.1F - l * 0.6F);
 		this.leftArm.posY = 0.1F - l * 0.6F;
 		this.rightArm.posX = (float) (-Math.PI / 2);
 		this.leftArm.posX = (float) (-Math.PI / 2);
 		this.rightArm.posX -= l * 1.2F - m * 0.4F;
 		this.leftArm.posX -= l * 1.2F - m * 0.4F;
-		this.rightArm.posZ = this.rightArm.posZ + MathHelper.cos(h * 0.09F) * 0.05F + 0.05F;
-		this.leftArm.posZ = this.leftArm.posZ - (MathHelper.cos(h * 0.09F) * 0.05F + 0.05F);
-		this.rightArm.posX = this.rightArm.posX + MathHelper.sin(h * 0.067F) * 0.05F;
-		this.leftArm.posX = this.leftArm.posX - MathHelper.sin(h * 0.067F) * 0.05F;
+		this.rightArm.posZ = this.rightArm.posZ + Mth.cos(h * 0.09F) * 0.05F + 0.05F;
+		this.leftArm.posZ = this.leftArm.posZ - (Mth.cos(h * 0.09F) * 0.05F + 0.05F);
+		this.rightArm.posX = this.rightArm.posX + Mth.sin(h * 0.067F) * 0.05F;
+		this.leftArm.posX = this.leftArm.posX - Mth.sin(h * 0.067F) * 0.05F;
 	}
 }
