--- BatEntityModel.java
+++ BatEntityModel.java
@@ -1,11 +1,11 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.passive.BatEntity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class BatEntityModel extends EntityModel {
 	private ModelPart head;
 	private ModelPart body;
 	private ModelPart rightWing;
@@ -76,13 +76,13 @@
 			this.head.posY = i / (180.0F / (float)Math.PI);
 			this.head.posZ = 0.0F;
 			this.head.setPivot(0.0F, 0.0F, 0.0F);
 			this.rightWing.setPivot(0.0F, 0.0F, 0.0F);
 			this.leftWing.setPivot(0.0F, 0.0F, 0.0F);
-			this.body.posX = (float) (Math.PI / 4) + MathHelper.cos(h * 0.1F) * 0.15F;
+			this.body.posX = (float) (Math.PI / 4) + Mth.cos(h * 0.1F) * 0.15F;
 			this.body.posY = 0.0F;
-			this.rightWing.posY = MathHelper.cos(h * 1.3F) * (float) Math.PI * 0.25F;
+			this.rightWing.posY = Mth.cos(h * 1.3F) * (float) Math.PI * 0.25F;
 			this.leftWing.posY = -this.rightWing.posY;
 			this.rightWingTip.posY = this.rightWing.posY * 0.5F;
 			this.leftWingTip.posY = -this.rightWing.posY * 0.5F;
 		}
 	}
