--- BlazeEntityModel.java
+++ BlazeEntityModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class BlazeEntityModel extends EntityModel {
 	private ModelPart[] rods = new ModelPart[12];
 	private ModelPart head;
 
@@ -31,31 +31,31 @@
 	@Override
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
 		float l = h * (float) Math.PI * -0.1F;
 
 		for (int m = 0; m < 4; m++) {
-			this.rods[m].pivotY = -2.0F + MathHelper.cos(((float)(m * 2) + h) * 0.25F);
-			this.rods[m].pivotX = MathHelper.cos(l) * 9.0F;
-			this.rods[m].pivotZ = MathHelper.sin(l) * 9.0F;
+			this.rods[m].pivotY = -2.0F + Mth.cos(((float)(m * 2) + h) * 0.25F);
+			this.rods[m].pivotX = Mth.cos(l) * 9.0F;
+			this.rods[m].pivotZ = Mth.sin(l) * 9.0F;
 			l++;
 		}
 
 		l = (float) (Math.PI / 4) + h * (float) Math.PI * 0.03F;
 
 		for (int m = 4; m < 8; m++) {
-			this.rods[m].pivotY = 2.0F + MathHelper.cos(((float)(m * 2) + h) * 0.25F);
-			this.rods[m].pivotX = MathHelper.cos(l) * 7.0F;
-			this.rods[m].pivotZ = MathHelper.sin(l) * 7.0F;
+			this.rods[m].pivotY = 2.0F + Mth.cos(((float)(m * 2) + h) * 0.25F);
+			this.rods[m].pivotX = Mth.cos(l) * 7.0F;
+			this.rods[m].pivotZ = Mth.sin(l) * 7.0F;
 			l++;
 		}
 
 		l = 0.47123894F + h * (float) Math.PI * -0.05F;
 
 		for (int m = 8; m < 12; m++) {
-			this.rods[m].pivotY = 11.0F + MathHelper.cos(((float)m * 1.5F + h) * 0.5F);
-			this.rods[m].pivotX = MathHelper.cos(l) * 5.0F;
-			this.rods[m].pivotZ = MathHelper.sin(l) * 5.0F;
+			this.rods[m].pivotY = 11.0F + Mth.cos(((float)m * 1.5F + h) * 0.5F);
+			this.rods[m].pivotX = Mth.cos(l) * 5.0F;
+			this.rods[m].pivotZ = Mth.sin(l) * 5.0F;
 			l++;
 		}
 
 		this.head.posY = i / (180.0F / (float)Math.PI);
 		this.head.posX = j / (180.0F / (float)Math.PI);
