--- BookModel.java
+++ BookModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class BookModel extends EntityModel {
 	public ModelPart leftCover = new ModelPart(this).setTextureOffset(0, 0).addCuboid(-6.0F, -5.0F, 0.0F, 6, 10, 0);
 	public ModelPart rightCover = new ModelPart(this).setTextureOffset(16, 0).addCuboid(0.0F, -5.0F, 0.0F, 6, 10, 0);
 	public ModelPart leftBlock;
@@ -35,18 +35,18 @@
 		this.rightPage.render(k);
 	}
 
 	@Override
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
-		float l = (MathHelper.sin(f * 0.02F) * 0.1F + 1.25F) * i;
+		float l = (Mth.sin(f * 0.02F) * 0.1F + 1.25F) * i;
 		this.leftCover.posY = (float) Math.PI + l;
 		this.rightCover.posY = -l;
 		this.leftBlock.posY = l;
 		this.rightBlock.posY = -l;
 		this.leftPage.posY = l - l * 2.0F * g;
 		this.rightPage.posY = l - l * 2.0F * h;
-		this.leftBlock.pivotX = MathHelper.sin(l);
-		this.rightBlock.pivotX = MathHelper.sin(l);
-		this.leftPage.pivotX = MathHelper.sin(l);
-		this.rightPage.pivotX = MathHelper.sin(l);
+		this.leftBlock.pivotX = Mth.sin(l);
+		this.rightBlock.pivotX = Mth.sin(l);
+		this.leftPage.pivotX = Mth.sin(l);
+		this.rightPage.pivotX = Mth.sin(l);
 	}
 }
