--- OcelotEntityModel.java
+++ OcelotEntityModel.java
@@ -1,13 +1,13 @@
 package net.minecraft.client.render.entity.model;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.passive.OcelotEntity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class OcelotEntityModel extends EntityModel {
 	ModelPart field_1496;
 	ModelPart field_1497;
 	ModelPart field_1498;
@@ -91,24 +91,24 @@
 		this.field_1502.posX = j / (180.0F / (float)Math.PI);
 		this.field_1502.posY = i / (180.0F / (float)Math.PI);
 		if (this.field_1504 != 3) {
 			this.field_1503.posX = (float) (Math.PI / 2);
 			if (this.field_1504 == 2) {
-				this.field_1496.posX = MathHelper.cos(f * 0.6662F) * 1.0F * g;
-				this.field_1497.posX = MathHelper.cos(f * 0.6662F + 0.3F) * 1.0F * g;
-				this.field_1498.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI + 0.3F) * 1.0F * g;
-				this.field_1499.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.0F * g;
-				this.field_1501.posX = 1.7278761F + (float) (Math.PI / 10) * MathHelper.cos(f) * g;
+				this.field_1496.posX = Mth.cos(f * 0.6662F) * 1.0F * g;
+				this.field_1497.posX = Mth.cos(f * 0.6662F + 0.3F) * 1.0F * g;
+				this.field_1498.posX = Mth.cos(f * 0.6662F + (float) Math.PI + 0.3F) * 1.0F * g;
+				this.field_1499.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.0F * g;
+				this.field_1501.posX = 1.7278761F + (float) (Math.PI / 10) * Mth.cos(f) * g;
 			} else {
-				this.field_1496.posX = MathHelper.cos(f * 0.6662F) * 1.0F * g;
-				this.field_1497.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.0F * g;
-				this.field_1498.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.0F * g;
-				this.field_1499.posX = MathHelper.cos(f * 0.6662F) * 1.0F * g;
+				this.field_1496.posX = Mth.cos(f * 0.6662F) * 1.0F * g;
+				this.field_1497.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.0F * g;
+				this.field_1498.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.0F * g;
+				this.field_1499.posX = Mth.cos(f * 0.6662F) * 1.0F * g;
 				if (this.field_1504 == 1) {
-					this.field_1501.posX = 1.7278761F + (float) (Math.PI / 4) * MathHelper.cos(f) * g;
+					this.field_1501.posX = 1.7278761F + (float) (Math.PI / 4) * Mth.cos(f) * g;
 				} else {
-					this.field_1501.posX = 1.7278761F + 0.47123894F * MathHelper.cos(f) * g;
+					this.field_1501.posX = 1.7278761F + 0.47123894F * Mth.cos(f) * g;
 				}
 			}
 		}
 	}
 
