--- QuadruPedEntityModel.java
+++ QuadruPedEntityModel.java
@@ -1,11 +1,11 @@
 package net.minecraft.client.render.entity.model;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class QuadruPedEntityModel extends EntityModel {
 	public ModelPart head;
 	public ModelPart torso;
 	public ModelPart backRightLeg;
@@ -68,11 +68,11 @@
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
 		float l = 180.0F / (float)Math.PI;
 		this.head.posX = j / (180.0F / (float)Math.PI);
 		this.head.posY = i / (180.0F / (float)Math.PI);
 		this.torso.posX = (float) (Math.PI / 2);
-		this.backRightLeg.posX = MathHelper.cos(f * 0.6662F) * 1.4F * g;
-		this.backLeftLeg.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
-		this.frontRightLeg.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
-		this.frontLeftLeg.posX = MathHelper.cos(f * 0.6662F) * 1.4F * g;
+		this.backRightLeg.posX = Mth.cos(f * 0.6662F) * 1.4F * g;
+		this.backLeftLeg.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
+		this.frontRightLeg.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
+		this.frontLeftLeg.posX = Mth.cos(f * 0.6662F) * 1.4F * g;
 	}
 }
