--- RabbitEntityModel.java
+++ RabbitEntityModel.java
@@ -1,13 +1,13 @@
 package net.minecraft.client.render.entity.model;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.passive.RabbitEntity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class RabbitEntityModel extends EntityModel {
 	ModelPart field_10568;
 	ModelPart field_10569;
 	ModelPart field_10570;
@@ -142,11 +142,11 @@
 		RabbitEntity rabbitEntity = (RabbitEntity)entity;
 		this.field_10579.posX = this.field_10575.posX = this.field_10576.posX = this.field_10577.posX = j * (float) (Math.PI / 180.0);
 		this.field_10579.posY = this.field_10575.posY = i * (float) (Math.PI / 180.0);
 		this.field_10576.posY = this.field_10579.posY - (float) (Math.PI / 12);
 		this.field_10577.posY = this.field_10579.posY + (float) (Math.PI / 12);
-		this.field_10580 = MathHelper.sin(rabbitEntity.getJumpProgress(l) * (float) Math.PI);
+		this.field_10580 = Mth.sin(rabbitEntity.getJumpProgress(l) * (float) Math.PI);
 		this.field_10570.posX = this.field_10571.posX = (this.field_10580 * 50.0F - 21.0F) * (float) (Math.PI / 180.0);
 		this.field_10568.posX = this.field_10569.posX = this.field_10580 * 50.0F * (float) (Math.PI / 180.0);
 		this.field_10573.posX = this.field_10574.posX = (this.field_10580 * -40.0F - 11.0F) * (float) (Math.PI / 180.0);
 	}
 
