--- SilverfishEntityModel.java
+++ SilverfishEntityModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class SilverfishEntityModel extends EntityModel {
 	private ModelPart[] field_1521;
 	private ModelPart[] field_1522;
 	private float[] field_1523 = new float[7];
@@ -51,12 +51,12 @@
 	}
 
 	@Override
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
 		for (int l = 0; l < this.field_1521.length; l++) {
-			this.field_1521[l].posY = MathHelper.cos(h * 0.9F + (float)l * 0.15F * (float) Math.PI) * (float) Math.PI * 0.05F * (float)(1 + Math.abs(l - 2));
-			this.field_1521[l].pivotX = MathHelper.sin(h * 0.9F + (float)l * 0.15F * (float) Math.PI) * (float) Math.PI * 0.2F * (float)Math.abs(l - 2);
+			this.field_1521[l].posY = Mth.cos(h * 0.9F + (float)l * 0.15F * (float) Math.PI) * (float) Math.PI * 0.05F * (float)(1 + Math.abs(l - 2));
+			this.field_1521[l].pivotX = Mth.sin(h * 0.9F + (float)l * 0.15F * (float) Math.PI) * (float) Math.PI * 0.2F * (float)Math.abs(l - 2);
 		}
 
 		this.field_1522[0].posY = this.field_1521[2].posY;
 		this.field_1522[1].posY = this.field_1521[4].posY;
 		this.field_1522[1].pivotX = this.field_1521[4].pivotX;
