--- SnowmanEntityModel.java
+++ SnowmanEntityModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class SnowmanEntityModel extends EntityModel {
 	public ModelPart field_1530;
 	public ModelPart field_1531;
 	public ModelPart field_1532;
@@ -35,12 +35,12 @@
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
 		super.setAngles(f, g, h, i, j, k, entity);
 		this.field_1532.posY = i / (180.0F / (float)Math.PI);
 		this.field_1532.posX = j / (180.0F / (float)Math.PI);
 		this.field_1530.posY = i / (180.0F / (float)Math.PI) * 0.25F;
-		float l = MathHelper.sin(this.field_1530.posY);
-		float m = MathHelper.cos(this.field_1530.posY);
+		float l = Mth.sin(this.field_1530.posY);
+		float m = Mth.cos(this.field_1530.posY);
 		this.field_1533.posZ = 1.0F;
 		this.field_1534.posZ = -1.0F;
 		this.field_1533.posY = 0.0F + this.field_1530.posY;
 		this.field_1534.posY = (float) Math.PI + this.field_1530.posY;
 		this.field_1533.pivotX = m * 5.0F;
