--- SpiderEntityModel.java
+++ SpiderEntityModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class SpiderEntityModel extends EntityModel {
 	public ModelPart field_1535;
 	public ModelPart field_1536;
 	public ModelPart field_1537;
@@ -92,18 +92,18 @@
 		this.field_1541.posY = -n * 1.0F - m;
 		this.field_1542.posY = -n * 1.0F + m;
 		this.field_1543.posY = n * 1.0F - m;
 		this.field_1544.posY = -n * 2.0F + m;
 		this.field_1545.posY = n * 2.0F - m;
-		float o = -(MathHelper.cos(f * 0.6662F * 2.0F + 0.0F) * 0.4F) * g;
-		float p = -(MathHelper.cos(f * 0.6662F * 2.0F + (float) Math.PI) * 0.4F) * g;
-		float q = -(MathHelper.cos(f * 0.6662F * 2.0F + (float) (Math.PI / 2)) * 0.4F) * g;
-		float r = -(MathHelper.cos(f * 0.6662F * 2.0F + (float) (Math.PI * 3.0 / 2.0)) * 0.4F) * g;
-		float s = Math.abs(MathHelper.sin(f * 0.6662F + 0.0F) * 0.4F) * g;
-		float t = Math.abs(MathHelper.sin(f * 0.6662F + (float) Math.PI) * 0.4F) * g;
-		float u = Math.abs(MathHelper.sin(f * 0.6662F + (float) (Math.PI / 2)) * 0.4F) * g;
-		float v = Math.abs(MathHelper.sin(f * 0.6662F + (float) (Math.PI * 3.0 / 2.0)) * 0.4F) * g;
+		float o = -(Mth.cos(f * 0.6662F * 2.0F + 0.0F) * 0.4F) * g;
+		float p = -(Mth.cos(f * 0.6662F * 2.0F + (float) Math.PI) * 0.4F) * g;
+		float q = -(Mth.cos(f * 0.6662F * 2.0F + (float) (Math.PI / 2)) * 0.4F) * g;
+		float r = -(Mth.cos(f * 0.6662F * 2.0F + (float) (Math.PI * 3.0 / 2.0)) * 0.4F) * g;
+		float s = Math.abs(Mth.sin(f * 0.6662F + 0.0F) * 0.4F) * g;
+		float t = Math.abs(Mth.sin(f * 0.6662F + (float) Math.PI) * 0.4F) * g;
+		float u = Math.abs(Mth.sin(f * 0.6662F + (float) (Math.PI / 2)) * 0.4F) * g;
+		float v = Math.abs(Mth.sin(f * 0.6662F + (float) (Math.PI * 3.0 / 2.0)) * 0.4F) * g;
 		this.field_1538.posY += o;
 		this.field_1539.posY += -o;
 		this.field_1540.posY += p;
 		this.field_1541.posY += -p;
 		this.field_1542.posY += q;
