--- VillagerEntityModel.java
+++ VillagerEntityModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class VillagerEntityModel extends EntityModel {
 	public ModelPart field_1557;
 	public ModelPart field_1558;
 	public ModelPart field_1559;
@@ -57,11 +57,11 @@
 		this.field_1557.posY = i / (180.0F / (float)Math.PI);
 		this.field_1557.posX = j / (180.0F / (float)Math.PI);
 		this.field_1559.pivotY = 3.0F;
 		this.field_1559.pivotZ = -1.0F;
 		this.field_1559.posX = -0.75F;
-		this.field_1560.posX = MathHelper.cos(f * 0.6662F) * 1.4F * g * 0.5F;
-		this.field_1561.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g * 0.5F;
+		this.field_1560.posX = Mth.cos(f * 0.6662F) * 1.4F * g * 0.5F;
+		this.field_1561.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g * 0.5F;
 		this.field_1560.posY = 0.0F;
 		this.field_1561.posY = 0.0F;
 	}
 }
