--- WitchEntityModel.java
+++ WitchEntityModel.java
@@ -1,10 +1,10 @@
 package net.minecraft.client.render.entity.model;
 
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class WitchEntityModel extends VillagerEntityModel {
 	public boolean field_5133;
 	private ModelPart field_5134 = new ModelPart(this).setTextureSize(64, 128);
 	private ModelPart field_5135;
@@ -41,13 +41,13 @@
 	@Override
 	public void setAngles(float f, float g, float h, float i, float j, float k, Entity entity) {
 		super.setAngles(f, g, h, i, j, k, entity);
 		this.field_5132.offsetX = this.field_5132.offsetY = this.field_5132.offsetZ = 0.0F;
 		float l = 0.01F * (float)(entity.getEntityId() % 10);
-		this.field_5132.posX = MathHelper.sin((float)entity.ticksAlive * l) * 4.5F * (float) Math.PI / 180.0F;
+		this.field_5132.posX = Mth.sin((float)entity.ticksAlive * l) * 4.5F * (float) Math.PI / 180.0F;
 		this.field_5132.posY = 0.0F;
-		this.field_5132.posZ = MathHelper.cos((float)entity.ticksAlive * l) * 2.5F * (float) Math.PI / 180.0F;
+		this.field_5132.posZ = Mth.cos((float)entity.ticksAlive * l) * 2.5F * (float) Math.PI / 180.0F;
 		if (this.field_5133) {
 			this.field_5132.posX = -0.9F;
 			this.field_5132.offsetZ = -0.09375F;
 			this.field_5132.offsetY = 0.1875F;
 		}
