--- WolfEntityModel.java
+++ WolfEntityModel.java
@@ -1,13 +1,13 @@
 package net.minecraft.client.render.entity.model;
 
-import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.legacy.GlStateManager;
 import net.minecraft.client.render.model.ModelPart;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.passive.WolfEntity;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class WolfEntityModel extends EntityModel {
 	public ModelPart field_1562;
 	public ModelPart field_1563;
 	public ModelPart field_1564;
@@ -86,11 +86,11 @@
 	public void animateModel(LivingEntity livingEntity, float f, float g, float h) {
 		WolfEntity wolfEntity = (WolfEntity)livingEntity;
 		if (wolfEntity.isAngry()) {
 			this.field_1568.posY = 0.0F;
 		} else {
-			this.field_1568.posY = MathHelper.cos(f * 0.6662F) * 1.4F * g;
+			this.field_1568.posY = Mth.cos(f * 0.6662F) * 1.4F * g;
 		}
 
 		if (wolfEntity.isSitting()) {
 			this.field_1569.setPivot(-1.0F, 16.0F, -3.0F);
 			this.field_1569.posX = (float) (Math.PI * 2.0 / 5.0);
@@ -114,14 +114,14 @@
 			this.field_1568.setPivot(-1.0F, 12.0F, 8.0F);
 			this.field_1564.setPivot(-2.5F, 16.0F, 7.0F);
 			this.field_1565.setPivot(0.5F, 16.0F, 7.0F);
 			this.field_1566.setPivot(-2.5F, 16.0F, -4.0F);
 			this.field_1567.setPivot(0.5F, 16.0F, -4.0F);
-			this.field_1564.posX = MathHelper.cos(f * 0.6662F) * 1.4F * g;
-			this.field_1565.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
-			this.field_1566.posX = MathHelper.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
-			this.field_1567.posX = MathHelper.cos(f * 0.6662F) * 1.4F * g;
+			this.field_1564.posX = Mth.cos(f * 0.6662F) * 1.4F * g;
+			this.field_1565.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
+			this.field_1566.posX = Mth.cos(f * 0.6662F + (float) Math.PI) * 1.4F * g;
+			this.field_1567.posX = Mth.cos(f * 0.6662F) * 1.4F * g;
 		}
 
 		this.field_1562.posZ = wolfEntity.method_2880(h) + wolfEntity.method_2876(h, 0.0F);
 		this.field_1569.posZ = wolfEntity.method_2876(h, -0.08F);
 		this.field_1563.posZ = wolfEntity.method_2876(h, -0.16F);
