--- Window.java
+++ Window.java
@@ -1,9 +1,9 @@
 package net.minecraft.client.util;
 
 import net.minecraft.client.MinecraftClient;
-import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.Mth;
 
 public class Window {
 	private final double scaledWidth;
 	private final double scaledHeight;
 	private int width;
@@ -28,12 +28,12 @@
 			this.scaleFactor--;
 		}
 
 		this.scaledWidth = (double)this.width / (double)this.scaleFactor;
 		this.scaledHeight = (double)this.height / (double)this.scaleFactor;
-		this.width = MathHelper.ceil(this.scaledWidth);
-		this.height = MathHelper.ceil(this.scaledHeight);
+		this.width = Mth.ceil(this.scaledWidth);
+		this.height = Mth.ceil(this.scaledHeight);
 	}
 
 	public int getWidth() {
 		return this.width;
 	}
